{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notebook for running literature search and validation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import json \n",
    "from Bio import Entrez\n",
    "import openai\n",
    "from utils.reference_checker import get_references_for_paragraphs, iter_dataframe\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "runVersion = 'initial'#'initial'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataType = \"Omics\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "if dataType == \"Omics\":\n",
    "    LLM_analysisFilePath  = 'data/omics_revamped_LLM_Enrichr_simVals_DF.tsv'\n",
    "    toSaveFilePath = 'data/omics_revamped_LLM_Enrichr_simVals_refs_2_DF' # remove the .tsv, because output will be saved as a dataframe(.tsv) and a dictionary (.json) in the function\n",
    "    jsonFilePath = 'jsonFiles/reference_checking_revision.json'\n",
    "    nameCol = 'GeneSetName'\n",
    "    LLM_analysisCol = 'LLM Analysis'\n",
    "\n",
    "elif dataType == \"GO_sets\":\n",
    "    LLM_analysisFilePath = 'data/GO_term_analysis/simrank_LLM_processed_selected_1000_go_terms.tsv'\n",
    "    toSaveFilePath = 'data/GO_term_analysis/simrank_LLM_processed_selected_1000_go_terms_refs'\n",
    "    jsonFilePath = 'reference_checking_task1.json'\n",
    "    examplesTORun = [\"GO:0019433\"] \n",
    "    nameCol = 'GO'\n",
    "    \n",
    "else:\n",
    "    print(\"Not implemented for dataType\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "if runVersion == 'additional':\n",
    "    LLM_analysisFilePath = toSaveFilePath + '.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'data/omics_revamped_LLM_Enrichr_simVals_DF.tsv'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LLM_analysisFilePath"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(jsonFilePath) as json_file:\n",
    "    config = json.load(json_file)\n",
    "\n",
    "openai.api_key = os.environ[\"OPENAI_API_KEY\"]\n",
    "email = config['EMAIL']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Read in the LLM analysis file\n",
    "df = pd.read_csv(LLM_analysisFilePath, sep='\\t', keep_default_na=False, na_values=['NaN'])\n",
    "df = df.replace({'None':None})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "## UPDATES 02/13/2023\n",
    "#iterate through the df and save df and dict\n",
    "\n",
    "res_df, res_dict = iter_dataframe(df, email, config, n=3, papers_query=20, verbose=False, return_paragraph_ref_data=False, id_col=nameCol, paragraph_col=LLM_analysisCol, runVersion=runVersion, save_path = toSaveFilePath)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "======================================="
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## use your own iterate dataframe loop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "runOnlyExamples = False\n",
    "examplesTORun = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Read in the LLM analysis file\n",
    "df = pd.read_csv(LLM_analysisFilePath, sep='\\t', keep_default_na=False, na_values=['NaN'])\n",
    "df = df.replace({'None':None})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if runVersion == 'initial':\n",
    "    df['referenced_analysis'] = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=========================================\n",
      "=========================================\n",
      "=========================================\n",
      "['dataframe row', 0]\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "1. CTRL, or carboxyl-terminal esterase/lipase, is involved in lipid metabolism, particularly in the hydrolysis of triglycerides and cholesteryl esters. This activity is crucial for lipid digestion and the subsequent absorption and processing within cells.\n",
      "===========================================================================\n",
      "187\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "1. CTRL, or carboxyl-terminal esterase/lipase, is involved in lipid metabolism, particularly in the hydrolysis of triglycerides and cholesteryl esters. This activity is crucial for lipid digestion and the subsequent absorption and processing within cells.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "CTRL\n",
      "301\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "1. CTRL, or carboxyl-terminal esterase/lipase, is involved in lipid metabolism, particularly in the hydrolysis of triglycerides and cholesteryl esters. This activity is crucial for lipid digestion and the subsequent absorption and processing within cells.\n",
      "    \n",
      "Result:\n",
      "lipid metabolism,hydrolysis,triglycerides\n",
      "PubMed Keywords:  (\"CTRL\") AND (\"lipid metabolism\" OR \"hydrolysis\" OR \"triglycerides\") AND (hasabstract[text])\n",
      "Serching paper with keywords...\n",
      "=========================================\n",
      "=========================================\n",
      "=========================================\n",
      "['dataframe row', 1]\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "1. LMF1 (Lipase Maturation Factor 1) is crucial for the proper folding and maturation of lipoprotein lipase (LPL) and hepatic lipase (HL), enzymes that are essential for lipid metabolism. LMF1 deficiency can lead to hypertriglyceridemia due to impaired lipase function.\n",
      "===========================================================================\n",
      "209\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "1. LMF1 (Lipase Maturation Factor 1) is crucial for the proper folding and maturation of lipoprotein lipase (LPL) and hepatic lipase (HL), enzymes that are essential for lipid metabolism. LMF1 deficiency can lead to hypertriglyceridemia due to impaired lipase function.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "LMF1,LPL,HL\n",
      "321\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "1. LMF1 (Lipase Maturation Factor 1) is crucial for the proper folding and maturation of lipoprotein lipase (LPL) and hepatic lipase (HL), enzymes that are essential for lipid metabolism. LMF1 deficiency can lead to hypertriglyceridemia due to impaired lipase function.\n",
      "    \n",
      "Result:\n",
      "lipid metabolism,lipase maturation,hypertriglyceridemia\n",
      "PubMed Keywords:  (\"LMF1\" OR \"LPL\" OR \"LIPC\") AND (\"lipid metabolism\" OR \"lipase maturation\" OR \"hypertriglyceridemia\") AND (hasabstract[text])\n",
      "Serching paper with keywords...\n",
      "18 references are queried\n",
      "182\n",
      "Lipoprotein Lipase and Its Regulators: An Unfolding Story. ['LMF1', 'LPL'] 50\n",
      "176\n",
      "Genetics of Hypertriglyceridemia. ['LMF1', 'LPL'] 32\n",
      "186\n",
      "Lipase maturation factor 1: a lipase chaperone involved in lipid metabolism. ['LMF1', 'LPL'] 30\n",
      "189\n",
      "GPIHBP1 and Lipoprotein Lipase, Partners in Plasma Triglyceride Metabolism. ['LPL'] 42\n",
      "181\n",
      "181\n",
      "178\n",
      "A Comprehensive Update on the Chylomicronemia Syndrome. ['LMF1'] 29\n",
      "183\n",
      "Regulation of lipoprotein lipase-mediated lipolysis of triglycerides. ['LPL'] 29\n",
      "186\n",
      "Triglyceride metabolism and angiopoietin-like proteins in lipoprotein lipase regulation. ['LPL'] 26\n",
      "185\n",
      "Lipase maturation factor 1: structure and role in lipase folding and assembly. ['LMF1'] 26\n",
      "180\n",
      "The Importance of Lipoprotein Lipase Regulation in Atherosclerosis. ['LPL'] 20\n",
      "191\n",
      "188\n",
      "Lipoprotein Lipase: Is It a Magic Target for the Treatment of Hypertriglyceridemia. ['LPL'] 8\n",
      "186\n",
      "Lipoprotein lipase transporter GPIHBP1 and triglyceride-rich lipoprotein metabolism. ['LPL'] 6\n",
      "175\n",
      "180\n",
      "Severe hypertriglyceridemia: Existing and emerging therapies. ['LPL'] 0\n",
      "201\n",
      "The number of title matching paper: 12\n",
      "357\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "1. LMF1 (Lipase Maturation Factor 1) is crucial for the proper folding and maturation of lipoprotein lipase (LPL) and hepatic lipase (HL), enzymes that are essential for lipid metabolism. LMF1 deficiency can lead to hypertriglyceridemia due to impaired lipase function.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Lipoprotein lipase (LPL) is one of the most important factors in systemic lipid partitioning and metabolism. It mediates intravascular hydrolysis of triglycerides packed in lipoproteins such as chylomicrons and very-low-density lipoprotein (VLDL). Since its initial discovery in the 1940s, its biology and pathophysiological significance have been well characterized. Nonetheless, several studies in the past decade, with recent delineation of LPL crystal structure and the discovery of several new regulators such as angiopoietin-like proteins (ANGPTLs), glycosylphosphatidylinositol-anchored high-density lipoprotein-binding protein 1 (GPIHBP1), lipase maturation factor 1 (LMF1) and Sel-1 suppressor of Lin-12-like 1 (SEL1L), have completely transformed our understanding of LPL biology.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  LMF1,LPL,LIPC\n",
      "Genetic Variants Associated with Severe Hypertriglyceridemia: LPL, APOC2, APOA5, GPIHBP1, LMF1, and APOE.\n",
      "Genetic Variants Associated with Severe Hypertriglyceridemia: LPL, APOC2, APOA5, GPIHBP1, LMF1, and APOE.\n",
      "528\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "1. LMF1 (Lipase Maturation Factor 1) is crucial for the proper folding and maturation of lipoprotein lipase (LPL) and hepatic lipase (HL), enzymes that are essential for lipid metabolism. LMF1 deficiency can lead to hypertriglyceridemia due to impaired lipase function.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Hypertriglyceridemia, a commonly encountered phenotype in cardiovascular and metabolic clinics, is surprisingly complex. A range of genetic variants, from single-nucleotide variants to large-scale copy number variants, can lead to either the severe or mild-to-moderate forms of the disease. At the genetic level, severely elevated triglyceride levels resulting from familial chylomicronemia syndrome (FCS) are caused by homozygous or biallelic loss-of-function variants in <i>LPL, APOC2, APOA5, LMF1</i>, and <i>GPIHBP1</i> genes. In contrast, susceptibility to multifactorial chylomicronemia (MCM), which has an estimated prevalence of ~1 in 600 and is at least 50-100-times more common than FCS, results from two different types of genetic variants: (1) rare heterozygous variants (minor allele frequency <1%) with variable penetrance in the five causal genes for FCS; and (2) common variants (minor allele frequency >5%) whose individually small phenotypic effects are quantified using a polygenic score. There is indirect evidence of similar complex genetic predisposition in other clinical phenotypes that have a component of hypertriglyceridemia, such as combined hyperlipidemia and dysbetalipoproteinemia. Future considerations include: (1) evaluation of whether the specific type of genetic predisposition to hypertriglyceridemia affects medical decisions or long-term outcomes; and (2) searching for other genetic contributors, including the role of genome-wide polygenic scores, novel genes, non-linear gene-gene or gene-environment interactions, and non-genomic mechanisms including epigenetics and mitochondrial DNA.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  LIPC\n",
      "396\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "1. LMF1 (Lipase Maturation Factor 1) is crucial for the proper folding and maturation of lipoprotein lipase (LPL) and hepatic lipase (HL), enzymes that are essential for lipid metabolism. LMF1 deficiency can lead to hypertriglyceridemia due to impaired lipase function.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Mutations in lipase maturation factor 1 (LMF1) are associated with severe hypertriglyceridemia in mice and human subjects. The underlying cause is impaired lipid clearance due to lipase deficiency. LMF1 is a chaperone of the endoplasmic reticulum (ER) and it is critically required for the post-translational activation of three vascular lipases: lipoprotein lipase (LPL), hepatic lipase (HL) and endothelial lipase (EL). As LMF1 is only required for the maturation of homodimeric, but not monomeric, lipases, it is likely involved in the assembly of inactive lipase subunits into active enzymes and/or the stabilization of active dimers. Herein, we provide an overview of current understanding of LMF1 function and propose that it may play a regulatory role in lipase activation and lipid metabolism. Further studies will be required to test this hypothesis and elucidate the full spectrum of phenotypes in combined lipase deficiency. This article is part of a Special Issue entitled Triglyceride Metabolism and Disease.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes\n",
      "Remained genes:  LIPC\n",
      "In paragraph 1, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "2. MFHAS1 (Malignant Fibrous Histiocytoma Amplified Sequence 1) is an oncogene with a role in macrophage activation and may be involved in lipid signaling pathways, although its precise biological function is not well characterized.\n",
      "===========================================================================\n",
      "188\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "2. MFHAS1 (Malignant Fibrous Histiocytoma Amplified Sequence 1) is an oncogene with a role in macrophage activation and may be involved in lipid signaling pathways, although its precise biological function is not well characterized.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "MFHAS1\n",
      "301\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "2. MFHAS1 (Malignant Fibrous Histiocytoma Amplified Sequence 1) is an oncogene with a role in macrophage activation and may be involved in lipid signaling pathways, although its precise biological function is not well characterized.\n",
      "    \n",
      "Result:\n",
      "oncogene,macrophage activation,lipid signaling\n",
      "PubMed Keywords:  (\"MFHAS1\") AND (\"oncogene\" OR \"macrophage activation\" OR \"lipid signaling\") AND (hasabstract[text])\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "172\n",
      "181\n",
      "172\n",
      "MFHAS1 promotes colorectal cancer progress by regulating polarization of tumor-associated macrophages via STAT6 signaling pathway. ['MFHAS1'] 19\n",
      "177\n",
      "MFHAS1 Is Associated with Sepsis and Stimulates TLR2/NF-κB Signaling Pathway Following Negative Regulation. ['MFHAS1'] 11\n",
      "165\n",
      "186\n",
      "185\n",
      "High Glucose Stimulates Expression of MFHAS1 to Mitigate Inflammation via Akt/HO-1 Pathway in Human Umbilical Vein Endothelial Cells. ['MFHAS1'] 4\n",
      "170\n",
      "Effects of MFHAS1 on cognitive impairment and dendritic pathology in the hippocampus of septic rats. ['MFHAS1'] 3\n",
      "189\n",
      "189\n",
      "187\n",
      "The number of title matching paper: 4\n",
      "516\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "2. MFHAS1 (Malignant Fibrous Histiocytoma Amplified Sequence 1) is an oncogene with a role in macrophage activation and may be involved in lipid signaling pathways, although its precise biological function is not well characterized.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Malignant fibrous histiocytoma amplified sequence 1 (MFHAS1) is a predicted oncoprotein that demonstrates tumorigenic activity in vivo; however, the mechanisms involved are unknown. Macrophages are divided into the pro-inflammatory M1 and anti-inflammatory/protumoral M2 subtypes. Tumor cells can induce M2 polarization of tumor-associated macrophages (TAMs) to promote metastasis; but the underlying pathways require to be elucidated. In this study, we detected a positive association between MFHAS1 expression in TAMs and human colorectal cancer (CRC) TNM stage. Supernatant of CT26 murine CRC cells induced MFHAS1 expression in RAW264.7 murine macrophages. Additionally, CT26 supernatant induced the M2 marker CD206 and activated the pro-M2 STAT6 and KLF4 signaling in control but not MFHAS1-silenced RAW264.7 macrophages. Moreover, supernatant of control, but not MFHAS1-silenced macrophages promoted CT26 cell proliferation, migration and epithelial-mesenchymal transition. Compared with control macrophages, MFHAS1-silenced macrophages showed significantly reduced protumoral effects in vivo. Together, these results suggested that CRC cells induce M2 polarization of TAMs through MFHAS1 induction and subsequent STAT6 and KLF4 activation to promote CRC progress. Finally, similar to CT26 supernatant stimulation, peroxisome proliferator-activated receptor-γ (PPARγ) activation by rosiglitazone induced M2 polarization of RAW264.7 macrophages through MFHAS1-dependent pathway. Our results highlight the role of MFHAS1 as a regulator of macrophages polarization and CRC progress.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  MFHAS1\n",
      "Genomic selection pressure discovery using site-frequency spectrum and reduced local variability statistics in Pakistani Dera-Din-Panah goat.\n",
      "477\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "2. MFHAS1 (Malignant Fibrous Histiocytoma Amplified Sequence 1) is an oncogene with a role in macrophage activation and may be involved in lipid signaling pathways, although its precise biological function is not well characterized.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Malignant fibrous histiocytoma amplified sequence 1 (MFHAS1) has a potential immunoregulatory role dependent on Toll-like receptors (TLRs). TLR2, associated with deleterious systemic inflammation, cardiac dysfunction, and acute kidney injury, acts synergistically in sepsis. The role of MFHAS1 in targeting TLR2 involved in sepsis has not been examined thus far. This study aimed to examine the relationship of MFHAS1 and sepsis, and the effect of MFHAS1 on the TLR2 signaling pathway. Blood samples were collected from eight sepsis patients after surgery and eight patients undergoing selective surgery to determine blood MFHAS1 levels. HEK 293 cells, RAW 264.7 macrophages and THP-1 monocytes were used to confirm the effect of MFHAS1 on TLR2 signaling pathway. Our study showed that blood MFHAS1 was significantly elevated in septic patients, and MFHAS1 was more increased in mononuclear cells from septic patients. Pam3CSK4 (TLR2 ligand) was found to induce MFHAS1 production in RAW 264.7 murine macrophages and THP-1 human monocytes in a time-dependent manner. MFHAS1 has dual effects on TLR2 signaling pathway and inflammation, i.e., inhibitory effect at 6 hours, and then stimulatory effect after 24 hours through the activation of TLR2/NF-κB signaling pathway, and MFHAS1 induced the phosphorylation of JNK and p38 after TLR2 stimulation.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "396\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "2. MFHAS1 (Malignant Fibrous Histiocytoma Amplified Sequence 1) is an oncogene with a role in macrophage activation and may be involved in lipid signaling pathways, although its precise biological function is not well characterized.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Hyperglycemia is a highly dangerous factor to various diseases, even resulting in death of people. Inflammation plays a key role in this process. The aim of this study was to explore the role of malignant fibrous histiocytoma amplified sequence 1 (MFHAS1) in high-glucose induced inflammation. Our research showed that high glucose stimulated the expression of MFHAS1, and overexpression of MFHAS1 can attenuate high-glucose induced inflammation in endothelial cells by decreasing the secretion of cytokines interleukin-1β (IL-1β), interleukin-1α (IL-1α), adhesion molecule intercellular adhesion molecule-1 (ICAM), interleukin-6 (IL-6), interleukin-8 (IL-8), and chemokine ligand 1 (CXCL-1). Furthermore, we found that MFHAS1 promoted the phosphorylation of Akt and the expression of heme oxygenase-1 (HO-1). Our results indicated that MFHAS1 deadened high-glucose induced inflammation by activating AKT/HO-1 pathway, suggesting that MFHAS1 may act as a new therapeutic target of diabetes mellitus.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 2, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "3. MR1 (Major Histocompatibility Complex, Class I-Related) presents microbial vitamin B metabolites to mucosal-associated invariant T (MAIT) cells. While not directly involved in lipid metabolism, it is part of the immune response in the gut where nutrient absorption and metabolism occur.\n",
      "===========================================================================\n",
      "193\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "3. MR1 (Major Histocompatibility Complex, Class I-Related) presents microbial vitamin B metabolites to mucosal-associated invariant T (MAIT) cells. While not directly involved in lipid metabolism, it is part of the immune response in the gut where nutrient absorption and metabolism occur.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "MR1\n",
      "306\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "3. MR1 (Major Histocompatibility Complex, Class I-Related) presents microbial vitamin B metabolites to mucosal-associated invariant T (MAIT) cells. While not directly involved in lipid metabolism, it is part of the immune response in the gut where nutrient absorption and metabolism occur.\n",
      "    \n",
      "Result:\n",
      "immune response,metabolite presentation,gut mucosa\n",
      "PubMed Keywords:  (\"MR1\") AND (\"immune response\" OR \"metabolite presentation\" OR \"gut mucosa\") AND (hasabstract[text])\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n"
     ]
    }
   ],
   "source": [
    "#####USE get_references_for_paragraph####\n",
    "for i, row in df.iterrows():\n",
    "#for i in range(startRow, df.shape[0]):\n",
    "#    row = df.iloc[i]\n",
    "  #  if runOnlyExamples: # Only run examples\n",
    "  #      if df.iloc[i][nameCol] not in examplesTORun: \n",
    "  #          continue\n",
    "    if runVersion == \"initial\":\n",
    "        if df.iloc[i]['referenced_analysis'] is not None:\n",
    "            continue\n",
    "    if runVersion == \"additional\":\n",
    "        if not (df.iloc[i]['referenced_analysis'] == ''):\n",
    "            continue # skip this row because already done\n",
    "    \n",
    "    print('=========================================')\n",
    "    print('=========================================')\n",
    "    print('=========================================')\n",
    "\n",
    "    print(['dataframe row', i])\n",
    "    # check out the llm analysis \n",
    "    example_analysis = df.iloc[i][LLM_analysisCol]\n",
    "    paragraphs = list(filter(lambda p: len(p.split()) > 5, example_analysis.split(\"\\n\")))\n",
    "    \n",
    "    try:\n",
    "        references, paragraph_dict = get_references_for_paragraphs(paragraphs, email = email, config =config, n=3, verbose=True, papers_query=20, return_paragraph_ref_data=True)\n",
    "        print(['Cannot get references for row', i])\n",
    "    except:\n",
    "        references = ''\n",
    "        \n",
    "    references.replace( '\\n', '')\n",
    "        \n",
    "    df.loc[i, 'referenced_analysis'] = references\n",
    "    if i%5==0:\n",
    "        df.to_csv(toSaveFilePath, sep = '\\t')\n",
    "        \n",
    "if not runOnlyExamples: \n",
    "    df.to_csv(toSaveFilePath, sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save MarkedParagraphs\n",
    "with open('data/MarkedParagraphs.pickle', 'wb') as f:\n",
    "    pickle.dump(MarkedParagraphs, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(toSaveFilePath, sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "llm_eval",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
