{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notebook for running literature search and validation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import json \n",
    "from Bio import Entrez\n",
    "import openai\n",
    "from utils.reference_checker import get_references_for_paragraphs, iter_dataframe\n",
    "import pickle\n",
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "runVersion = 'initial'#'initial'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataType = \"Omics\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "if dataType == \"Omics\":\n",
    "    LLM_analysisFilePath  = 'data/omics_revamped_LLM_Enrichr_simVals_DF.tsv'\n",
    "    toSaveFilePath = 'data/omics_revamped_LLM_Enrichr_simVals_refs_2_DF' # remove the .tsv, because output will be saved as a dataframe(.tsv) and a dictionary (.json) in the function\n",
    "    jsonFilePath = 'jsonFiles/reference_checking_revision.json'\n",
    "    nameCol = 'GeneSetName'\n",
    "    LLM_analysisCol = 'LLM Analysis'\n",
    "\n",
    "elif dataType == \"GO_sets\":\n",
    "    LLM_analysisFilePath = 'data/GO_term_analysis/simrank_LLM_processed_selected_1000_go_terms.tsv'\n",
    "    toSaveFilePath = 'data/GO_term_analysis/simrank_LLM_processed_selected_1000_go_terms_refs'\n",
    "    jsonFilePath = 'reference_checking_task1.json'\n",
    "    examplesTORun = [\"GO:0019433\"] \n",
    "    nameCol = 'GO'\n",
    "    \n",
    "else:\n",
    "    print(\"Not implemented for dataType\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "if runVersion == 'additional':\n",
    "    LLM_analysisFilePath = toSaveFilePath + '.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'data/omics_revamped_LLM_Enrichr_simVals_DF.tsv'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LLM_analysisFilePath"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(jsonFilePath) as json_file:\n",
    "    config = json.load(json_file)\n",
    "\n",
    "openai.api_key = os.environ[\"OPENAI_API_KEY\"]\n",
    "email = config['EMAIL']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Read in the LLM analysis file\n",
    "df = pd.read_csv(LLM_analysisFilePath, sep='\\t', keep_default_na=False, na_values=['NaN'])\n",
    "df = df.replace({'None':None})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Unnamed: 0.1  Unnamed: 0 Source    GeneSetID  GeneSetName  \\\n",
      "0             0           0   NeST  Cluster1-10  Cluster1-10   \n",
      "1             1           0   NeST  Cluster1-11  Cluster1-11   \n",
      "\n",
      "                                            GeneList  n_Genes  \\\n",
      "0  CTRL HSD17B14 KIAA0232 PAQR8 PLA2G1B RNF145 SG...       12   \n",
      "1  LMF1 MFHAS1 MR1 PLA2G1B RASL11A RNF145 SLC2A6 ...       12   \n",
      "\n",
      "                                    LLM Name  \\\n",
      "0     Lipid Metabolism and Membrane Dynamics   \n",
      "1  Lipid Metabolism and Membrane Trafficking   \n",
      "\n",
      "                                        LLM Analysis  Score  Rank Overlap  \\\n",
      "0  1. CTRL, or carboxyl-terminal esterase/lipase,...   0.85     0    2/21   \n",
      "1  1. LMF1 (Lipase Maturation Factor 1) is crucia...   0.85     0    2/85   \n",
      "\n",
      "    P-value  Adjusted P-value             Genes  \\\n",
      "0  0.000069          0.008264  PLA2G1B;HSD17B14   \n",
      "1  0.001146          0.050873       PLA2G1B;MR1   \n",
      "\n",
      "                                                Term  \\\n",
      "0               Lipid Catabolic Process (GO:0016042)   \n",
      "1  Defense Response To Gram-positive Bacterium (G...   \n",
      "\n",
      "                                        GO term       GO ID  \\\n",
      "0                      Lipid Catabolic Process   GO:0016042   \n",
      "1  Defense Response To Gram-positive Bacterium   GO:0050830   \n",
      "\n",
      "                                       GO_term_genes  LLM_name_GO_term_sim  \n",
      "0  CES1 PNLIPRP1 PLA2G1B CES1P1 AOAH HSD17B14 APO...              0.603573  \n",
      "1  GBP7 CHGA GBP6 GSDMD DEFB104A H2BC8 RNASE12 RN...              0.317063  \n"
     ]
    }
   ],
   "source": [
    "# ### comment this cell if running the full dataset, this is for a toy example\n",
    "\n",
    "# df = df.head(2) #only take the first 2 rows as a toy example\n",
    "# print(df)\n",
    "# # showing how to run a toy example\n",
    "# toSaveFilePath = 'data/omics_toy' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "187\n",
      "301\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "166\n",
      "165\n",
      "171\n",
      "173\n",
      "163\n",
      "167\n",
      "176\n",
      "201\n",
      "167\n",
      "194\n",
      "194\n",
      "169\n",
      "180\n",
      "The number of title matching paper: 1\n",
      "551\n",
      "In paragraph 1, 0 references are matched\n",
      "\n",
      "\n",
      "204\n",
      "314\n",
      "Serching paper with keywords...\n",
      "6 references are queried\n",
      "193\n",
      "205\n",
      "184\n",
      "202\n",
      "195\n",
      "The number of title matching paper: 2\n",
      "457\n",
      "Development of a clinical prediction model for diabetic kidney disease with glucose and lipid metabolism disorders based on machine learning and bioinformatics technology.\n",
      "405\n",
      "In paragraph 2, 2 references are matched\n",
      "\n",
      "\n",
      "176\n",
      "285\n",
      "Serching paper with keywords...\n",
      "1 references are queried\n",
      "156\n",
      "The number of title matching paper: 0\n",
      "In paragraph 3, 0 references are matched\n",
      "\n",
      "\n",
      "174\n",
      "286\n",
      "Serching paper with keywords...\n",
      "2 references are queried\n",
      "151\n",
      "163\n",
      "The number of title matching paper: 1\n",
      "609\n",
      "Expression of membrane progesterone receptors (mPR/PAQR) in ovarian cancer cells: implications for progesterone-induced signaling events.\n",
      "In paragraph 4, 1 references are matched\n",
      "\n",
      "\n",
      "196\n",
      "308\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "171\n",
      "184\n",
      "179\n",
      "182\n",
      "177\n",
      "325\n",
      "179\n",
      "181\n",
      "179\n",
      "182\n",
      "180\n",
      "174\n",
      "182\n",
      "The number of title matching paper: 6\n",
      "448\n",
      "Kinetics of group IB and IIA phospholipase A2 during low-volume continuous hemodiafiltration in severe acute pancreatitis.\n",
      "498\n",
      "406\n",
      "In paragraph 5, 3 references are matched\n",
      "\n",
      "\n",
      "172\n",
      "285\n",
      "Serching paper with keywords...\n",
      "5 references are queried\n",
      "167\n",
      "163\n",
      "157\n",
      "152\n",
      "The number of title matching paper: 4\n",
      "463\n",
      "The E3 ubiquitin ligase RNF144B is LPS-inducible in human, but not mouse, macrophages and promotes inducible IL-1β expression.\n",
      "508\n",
      "469\n",
      "In paragraph 6, 3 references are matched\n",
      "\n",
      "\n",
      "195\n",
      "310\n",
      "Serching paper with keywords...\n",
      "16 references are queried\n",
      "173\n",
      "176\n",
      "186\n",
      "175\n",
      "174\n",
      "The number of title matching paper: 3\n",
      "427\n",
      "Cryo-EM structure of human sphingomyelin synthase and its mechanistic implications for sphingomyelin synthesis.\n",
      "600\n",
      "408\n",
      "In paragraph 7, 3 references are matched\n",
      "\n",
      "\n",
      "192\n",
      "303\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "170\n",
      "184\n",
      "185\n",
      "175\n",
      "186\n",
      "173\n",
      "175\n",
      "185\n",
      "219\n",
      "168\n",
      "The number of title matching paper: 4\n",
      "618\n",
      "Protein kinase C activation upregulates human L-type amino acid transporter 2 function.\n",
      "388\n",
      "566\n",
      "In paragraph 8, 3 references are matched\n",
      "\n",
      "\n",
      "184\n",
      "293\n",
      "Serching paper with keywords...\n",
      "No results found.\n",
      "0 references are queried\n",
      "No paper searched!!\n",
      "The number of title matching paper: 0\n",
      "In paragraph 9, 0 references are matched\n",
      "\n",
      "\n",
      "175\n",
      "287\n",
      "Serching paper with keywords...\n",
      "No results found.\n",
      "0 references are queried\n",
      "No paper searched!!\n",
      "The number of title matching paper: 0\n",
      "In paragraph 10, 0 references are matched\n",
      "\n",
      "\n",
      "178\n",
      "292\n",
      "Serching paper with keywords...\n",
      "4 references are queried\n",
      "156\n",
      "166\n",
      "175\n",
      "169\n",
      "The number of title matching paper: 2\n",
      "465\n",
      "[Ubiquitin-conjugating enzyme UBE2Q2 participates in HUWE1-mediated protection on renal tubulointerstitial fibrosis].\n",
      "579\n",
      "In paragraph 11, 2 references are matched\n",
      "\n",
      "\n",
      "172\n",
      "283\n",
      "Serching paper with keywords...\n",
      "1 references are queried\n",
      "157\n",
      "The number of title matching paper: 0\n",
      "In paragraph 12, 0 references are matched\n",
      "\n",
      "\n",
      "204\n",
      "318\n",
      "No keyword generated skip referencing\n",
      "In paragraph 13, 0 references are matched\n",
      "\n",
      "\n",
      "Total 17 references are queried\n",
      "[[], ['Qureshi, Rehana, Picon-Ruiz, Manuel, Aurrekoetxea-Rodriguez, Iskander, Nunes de Paiva, Vanessa, D\\'Amico, Massimo, Yoon, Hyunho, Radhakrishnan, Ramya, Morata-Tarifa, Cynthia, Ince, Tan, Lippman, Marc E, Thaller, Seth R, Rodgers, Steven E, Kesmodel, Susan, Vivanco, Maria Del Mar, Slingerland, Joyce M. \"The Major Pre- and Postmenopausal Estrogens Play Opposing Roles in Obesity-Driven Mammary Inflammation and Breast Cancer Development.\" Cell metabolism, 2020, pp. 1154-1172.e9, doi: https://doi.org/10.1016/j.cmet.2020.05.008', 'Sivik, T, Vikingsson, S, Gréen, H, Jansson, A. \"Expression patterns of 17β-hydroxysteroid dehydrogenase 14 in human tissues.\" Hormone and metabolic research = Hormon- und Stoffwechselforschung = Hormones et metabolisme, 2012, pp. 949-56, doi: https://doi.org/10.1055/s-0032-1321815'], [], ['Charles, Nathan J, Thomas, Peter, Lange, Carol A. \"Expression of membrane progesterone receptors (mPR/PAQR) in ovarian cancer cells: implications for progesterone-induced signaling events.\" Hormones & cancer, 2010, pp. 167-76, doi: https://doi.org/10.1007/s12672-010-0023-9'], ['Labonté, Eric D, Kirby, R Jason, Schildmeyer, Nicholas M, Cannon, April M, Huggins, Kevin W, Hui, David Y. \"Group 1B phospholipase A2-mediated lysophospholipid absorption directly contributes to postprandial hyperglycemia.\" Diabetes, 2006, pp. 935-41, doi: https://doi.org/10.2337/diabetes.55.04.06.db05-1286', 'Hui, D Y, Cope, M J, Labonté, E D, Chang, H-T, Shao, J, Goka, E, Abousalham, A, Charmot, D, Buysse, J. \"The phospholipase A(2) inhibitor methyl indoxam suppresses diet-induced obesity and glucose intolerance in mice.\" British journal of pharmacology, 2009, pp. 1263-9, doi: https://doi.org/10.1111/j.1476-5381.2009.00308.x', 'Hui, David Y. \"Group 1B phospholipase A<sub>2</sub> in metabolic and inflammatory disease modulation.\" Biochimica et biophysica acta. Molecular and cell biology of lipids, 2019, pp. 784-788, doi: https://doi.org/10.1016/j.bbalip.2018.07.001'], ['Menzies, Sam A, Volkmar, Norbert, van den Boomen, Dick Jh, Timms, Richard T, Dickson, Anna S, Nathan, James A, Lehner, Paul J. \"The sterol-responsive RNF145 E3 ubiquitin ligase mediates the degradation of HMG-CoA reductase together with gp78 and Hrd1.\" eLife, 2018, pp.  , doi: https://doi.org/10.7554/eLife.40009', 'Jiang, Lu-Yi, Jiang, Wei, Tian, Na, Xiong, Yan-Ni, Liu, Jie, Wei, Jian, Wu, Kai-Yue, Luo, Jie, Shi, Xiong-Jie, Song, Bao-Liang. \"Ring finger protein 145 (RNF145) is a ubiquitin ligase for sterol-induced degradation of HMG-CoA reductase.\" The Journal of biological chemistry, 2018, pp. 4047-4055, doi: https://doi.org/10.1074/jbc.RA117.001260', 'Cook, Emma C L, Nelson, Jessica K, Sorrentino, Vincenzo, Koenis, Duco, Moeton, Martina, Scheij, Saskia, Ottenhoff, Roelof, Bleijlevens, Boris, Loregger, Anke, Zelcer, Noam. \"Identification of the ER-resident E3 ubiquitin ligase RNF145 as a novel LXR-regulated gene.\" PloS one, 2017, pp. e0172721, doi: https://doi.org/10.1371/journal.pone.0172721'], ['Hirata, Yuichi, Ikeda, Kazutaka, Sudoh, Masayuki, Tokunaga, Yuko, Suzuki, Akemi, Weng, Leiyun, Ohta, Masatoshi, Tobita, Yoshimi, Okano, Ken, Ozeki, Kazuhisa, Kawasaki, Kenichi, Tsukuda, Takuo, Katsume, Asao, Aoki, Yuko, Umehara, Takuya, Sekiguchi, Satoshi, Toyoda, Tetsuya, Shimotohno, Kunitada, Soga, Tomoyoshi, Nishijima, Masahiro, Taguchi, Ryo, Kohara, Michinori. \"Self-enhancement of hepatitis C virus replication by promotion of specific sphingolipid biosynthesis.\" PLoS pathogens, 2012, pp. e1002860, doi: https://doi.org/10.1371/journal.ppat.1002860', 'Jęśko, Henryk, Wencel, Przemysław L, Wójtowicz, Sylwia, Strosznajder, Joanna, Lukiw, Walter J, Strosznajder, Robert P. \"Fingolimod Affects Transcription of Genes Encoding Enzymes of Ceramide Metabolism in Animal Model of Alzheimer\\'s Disease.\" Molecular neurobiology, 2020, pp. 2799-2811, doi: https://doi.org/10.1007/s12035-020-01908-3', 'Komuro, Mariko, Nagane, Masaki, Fukuyama, Tomoki, Luo, Xiaolin, Hiraki, Shinobu, Miyanabe, Masakatsu, Ishikawa, Miyuki, Niwa, Chiaki, Murakami, Hironobu, Okamoto, Mariko, Yamashita, Tadashi. \"Sphingomyelin maintains the cutaneous barrier via regulation of the STAT3 pathway.\" FASEB journal : official publication of the Federation of American Societies for Experimental Biology, 2022, pp. e22111, doi: https://doi.org/10.1096/fj.202100721RR'], ['Gaccioli, Francesca, Aye, Irving L M H, Roos, Sara, Lager, Susanne, Ramirez, Vanessa I, Kanai, Yoshikatsu, Powell, Theresa L, Jansson, Thomas. \"Expression and functional characterisation of System L amino acid transporters in the human term placenta.\" Reproductive biology and endocrinology : RB&E, 2015, pp. 57, doi: https://doi.org/10.1186/s12958-015-0054-8', 'Espino Guarch, Meritxell, Font-Llitjós, Mariona, Murillo-Cuesta, Silvia, Errasti-Murugarren, Ekaitz, Celaya, Adelaida M, Girotto, Giorgia, Vuckovic, Dragana, Mezzavilla, Massimo, Vilches, Clara, Bodoy, Susanna, Sahún, Ignasi, González, Laura, Prat, Esther, Zorzano, Antonio, Dierssen, Mara, Varela-Nieto, Isabel, Gasparini, Paolo, Palacín, Manuel, Nunes, Virginia. \"Mutations in L-type amino acid transporter-2 support <i>SLC7A8</i> as a novel gene involved in age-related hearing loss.\" eLife, 2018, pp.  , doi: https://doi.org/10.7554/eLife.31511', 'Zaugg, Jonas, Huang, Xiao, Ziegler, Fabian, Rubin, Matthias, Graff, Julien, Müller, Jennifer, Moser-Hässig, Ruedi, Powell, Theresa, Gertsch, Jürg, Altmann, Karl-Heinz, Albrecht, Christiane. \"Small molecule inhibitors provide insights into the relevance of LAT1 and LAT2 in materno-foetal amino acid transport.\" Journal of cellular and molecular medicine, 2020, pp. 12681-12693, doi: https://doi.org/10.1111/jcmm.15840'], [], [], ['Nikseresht, Mohsen, Seghatoleslam, Atefeh, Monabati, Ahmad, Talei, Abdolrassul, Ghalati, Farzaneh Bozorg, Owji, Ali Akbar. \"Overexpression of the novel human gene, UBE2Q2, in breast cancer.\" Cancer genetics and cytogenetics, 2010, pp. 101-6, doi: https://doi.org/10.1016/j.cancergencyto.2009.11.020', 'Wang, Zheng, Dong, Hao, Li, Min, Liang, Xiu-Bin. \"[Ubiquitin-conjugating enzyme UBE2Q2 participates in HUWE1-mediated protection on renal tubulointerstitial fibrosis].\" Sheng li xue bao : [Acta physiologica Sinica], 2022, pp. 117-124.'], [], []]\n",
      "209\n",
      "321\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "182\n",
      "176\n",
      "179\n",
      "179\n",
      "177\n",
      "189\n",
      "181\n",
      "183\n",
      "181\n",
      "178\n",
      "177\n",
      "191\n",
      "177\n",
      "180\n",
      "186\n",
      "174\n",
      "190\n",
      "175\n",
      "180\n",
      "The number of title matching paper: 12\n",
      "357\n",
      "Medical management of hypertriglyceridemia in pancreatitis.\n",
      "Medical management of hypertriglyceridemia in pancreatitis.\n",
      "528\n",
      "513\n",
      "In paragraph 1, 3 references are matched\n",
      "\n",
      "\n",
      "188\n",
      "301\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "172\n",
      "182\n",
      "172\n",
      "177\n",
      "165\n",
      "187\n",
      "185\n",
      "189\n",
      "190\n",
      "187\n",
      "180\n",
      "The number of title matching paper: 6\n",
      "429\n",
      "Genomic selection pressure discovery using site-frequency spectrum and reduced local variability statistics in Pakistani Dera-Din-Panah goat.\n",
      "516\n",
      "477\n",
      "In paragraph 2, 3 references are matched\n",
      "\n",
      "\n",
      "193\n",
      "305\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "165\n",
      "162\n",
      "161\n",
      "162\n",
      "166\n",
      "162\n",
      "168\n",
      "163\n",
      "168\n",
      "164\n",
      "164\n",
      "162\n",
      "163\n",
      "163\n",
      "170\n",
      "172\n",
      "176\n",
      "174\n",
      "The number of title matching paper: 14\n",
      "327\n",
      "Role of MR1-driven signals and amphiregulin on the recruitment and repair function of MAIT cells during skin wound healing.\n",
      "439\n",
      "308\n",
      "In paragraph 3, 3 references are matched\n",
      "\n",
      "\n",
      "193\n",
      "302\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "181\n",
      "176\n",
      "201\n",
      "168\n",
      "174\n",
      "175\n",
      "176\n",
      "178\n",
      "176\n",
      "179\n",
      "177\n",
      "171\n",
      "179\n",
      "The number of title matching paper: 4\n",
      "445\n",
      "Kinetics of group IB and IIA phospholipase A2 during low-volume continuous hemodiafiltration in severe acute pancreatitis.\n",
      "403\n",
      "506\n",
      "In paragraph 4, 3 references are matched\n",
      "\n",
      "\n",
      "192\n",
      "304\n",
      "Serching paper with keywords...\n",
      "1 references are queried\n",
      "The number of title matching paper: 0\n",
      "In paragraph 5, 0 references are matched\n",
      "\n",
      "\n",
      "182\n",
      "295\n",
      "Serching paper with keywords...\n",
      "4 references are queried\n",
      "177\n",
      "173\n",
      "167\n",
      "The number of title matching paper: 3\n",
      "473\n",
      "Ubiquitin-mediated regulation of sterol homeostasis.\n",
      "518\n",
      "479\n",
      "In paragraph 6, 3 references are matched\n",
      "\n",
      "\n",
      "190\n",
      "304\n",
      "Serching paper with keywords...\n",
      "15 references are queried\n",
      "190\n",
      "165\n",
      "170\n",
      "173\n",
      "166\n",
      "179\n",
      "177\n",
      "167\n",
      "182\n",
      "The number of title matching paper: 1\n",
      "558\n",
      "Identification and development of a five-gene signature to improve the prediction of mechanical ventilator-free days for patients with COVID-19.\n",
      "In paragraph 7, 1 references are matched\n",
      "\n",
      "\n",
      "181\n",
      "292\n",
      "Serching paper with keywords...\n",
      "3 references are queried\n",
      "167\n",
      "168\n",
      "170\n",
      "The number of title matching paper: 3\n",
      "437\n",
      "Structural Studies Reveal Unique Non-canonical Regulators of G Protein Signaling Homology (RH) Domains in Sorting Nexins.\n",
      "409\n",
      "369\n",
      "In paragraph 8, 3 references are matched\n",
      "\n",
      "\n",
      "179\n",
      "291\n",
      "Serching paper with keywords...\n",
      "3 references are queried\n",
      "The number of title matching paper: 0\n",
      "In paragraph 9, 0 references are matched\n",
      "\n",
      "\n",
      "183\n",
      "298\n",
      "Serching paper with keywords...\n",
      "No results found.\n",
      "0 references are queried\n",
      "No paper searched!!\n",
      "The number of title matching paper: 0\n",
      "In paragraph 10, 0 references are matched\n",
      "\n",
      "\n",
      "180\n",
      "293\n",
      "Serching paper with keywords...\n",
      "6 references are queried\n",
      "176\n",
      "159\n",
      "165\n",
      "158\n",
      "173\n",
      "The number of title matching paper: 3\n",
      "444\n",
      "[Predictive value of deubiquitination enzymes USP37 in the prognosis of breast cancer].\n",
      "390\n",
      "464\n",
      "In paragraph 11, 3 references are matched\n",
      "\n",
      "\n",
      "178\n",
      "288\n",
      "Serching paper with keywords...\n",
      "7 references are queried\n",
      "163\n",
      "165\n",
      "163\n",
      "174\n",
      "157\n",
      "164\n",
      "152\n",
      "The number of title matching paper: 2\n",
      "377\n",
      "Structural insights into ORF10 recognition by ZYG11B.\n",
      "299\n",
      "In paragraph 12, 2 references are matched\n",
      "\n",
      "\n",
      "236\n",
      "349\n",
      "No keyword generated skip referencing\n",
      "In paragraph 13, 0 references are matched\n",
      "\n",
      "\n",
      "Total 24 references are queried\n",
      "[['Wu, Shuangcheng Alivia, Kersten, Sander, Qi, Ling. \"Lipoprotein Lipase and Its Regulators: An Unfolding Story.\" Trends in endocrinology and metabolism: TEM, 2021, pp. 48-61, doi: https://doi.org/10.1016/j.tem.2020.11.005', 'Dron, Jacqueline S, Hegele, Robert A. \"Genetics of Hypertriglyceridemia.\" Frontiers in endocrinology, 2020, pp. 455, doi: https://doi.org/10.3389/fendo.2020.00455', 'Alves, Mara, Laranjeira, Francisco, Correia-da-Silva, Georgina. \"Understanding Hypertriglyceridemia: Integrating Genetic Insights.\" Genes, 2024, pp.  , doi: https://doi.org/10.3390/genes15020190'], ['Zhong, Jing, Wang, Huihui, Chen, Wankun, Sun, Zhirong, Chen, Jiawei, Xu, Yajun, Weng, Meilin, Shi, Qiqing, Ma, Duan, Miao, Changhong. \"Ubiquitylation of MFHAS1 by the ubiquitin ligase praja2 promotes M1 macrophage polarization by activating JNK and p38 pathways.\" Cell death & disease, 2017, pp. e2763, doi: https://doi.org/10.1038/cddis.2017.102', 'Chen, Wankun, Xu, Yajun, Zhong, Jing, Wang, Huihui, Weng, Meilin, Cheng, Qian, Wu, Qichao, Sun, Zhirong, Jiang, Hui, Zhu, Minmin, Ren, Yu, Xu, Pingbo, Chen, Jiawei, Miao, Changhong. \"MFHAS1 promotes colorectal cancer progress by regulating polarization of tumor-associated macrophages via STAT6 signaling pathway.\" Oncotarget, 2016, pp. 78726-78735, doi: https://doi.org/10.18632/oncotarget.12807', 'Zhong, Jing, Shi, Qi-Qing, Zhu, Min-Min, Shen, Jian, Wang, Hui-Hui, Ma, Duan, Miao, Chang-Hong. \"MFHAS1 Is Associated with Sepsis and Stimulates TLR2/NF-κB Signaling Pathway Following Negative Regulation.\" PloS one, 2015, pp. e0143662, doi: https://doi.org/10.1371/journal.pone.0143662'], ['Godfrey, Dale I, Koay, Hui-Fern, McCluskey, James, Gherardin, Nicholas A. \"The biology and functional importance of MAIT cells.\" Nature immunology, 2019, pp. 1110-1128, doi: https://doi.org/10.1038/s41590-019-0444-8', 'Hinks, Timothy S C, Zhang, Xia-Wei. \"MAIT Cell Activation and Functions.\" Frontiers in immunology, 2020, pp. 1014, doi: https://doi.org/10.3389/fimmu.2020.01014', 'Meermeier, Erin W, Harriff, Melanie J, Karamooz, Elham, Lewinsohn, David M. \"MAIT cells and microbial immunity.\" Immunology and cell biology, 2018, pp. 607-617, doi: https://doi.org/10.1111/imcb.12022'], ['Labonté, Eric D, Kirby, R Jason, Schildmeyer, Nicholas M, Cannon, April M, Huggins, Kevin W, Hui, David Y. \"Group 1B phospholipase A2-mediated lysophospholipid absorption directly contributes to postprandial hyperglycemia.\" Diabetes, 2006, pp. 935-41, doi: https://doi.org/10.2337/diabetes.55.04.06.db05-1286', 'Hui, David Y. \"Group 1B phospholipase A<sub>2</sub> in metabolic and inflammatory disease modulation.\" Biochimica et biophysica acta. Molecular and cell biology of lipids, 2019, pp. 784-788, doi: https://doi.org/10.1016/j.bbalip.2018.07.001', 'Ewers, Maren, Epple, Denise, Bugert, Peter, Rosendahl, Jonas, Witt, Heiko. \"Genetic analysis of pancreatic phospholipase A2 (PLA2G1B) in patients with chronic pancreatitis.\" Pancreatology : official journal of the International Association of Pancreatology (IAP) ... [et al.], 2022, pp. 244-247, doi: https://doi.org/10.1016/j.pan.2022.01.003'], [], ['Menzies, Sam A, Volkmar, Norbert, van den Boomen, Dick Jh, Timms, Richard T, Dickson, Anna S, Nathan, James A, Lehner, Paul J. \"The sterol-responsive RNF145 E3 ubiquitin ligase mediates the degradation of HMG-CoA reductase together with gp78 and Hrd1.\" eLife, 2018, pp.  , doi: https://doi.org/10.7554/eLife.40009', 'Jiang, Lu-Yi, Jiang, Wei, Tian, Na, Xiong, Yan-Ni, Liu, Jie, Wei, Jian, Wu, Kai-Yue, Luo, Jie, Shi, Xiong-Jie, Song, Bao-Liang. \"Ring finger protein 145 (RNF145) is a ubiquitin ligase for sterol-induced degradation of HMG-CoA reductase.\" The Journal of biological chemistry, 2018, pp. 4047-4055, doi: https://doi.org/10.1074/jbc.RA117.001260', 'Cook, Emma C L, Nelson, Jessica K, Sorrentino, Vincenzo, Koenis, Duco, Moeton, Martina, Scheij, Saskia, Ottenhoff, Roelof, Bleijlevens, Boris, Loregger, Anke, Zelcer, Noam. \"Identification of the ER-resident E3 ubiquitin ligase RNF145 as a novel LXR-regulated gene.\" PloS one, 2017, pp. e0172721, doi: https://doi.org/10.1371/journal.pone.0172721'], ['Chen, Sing-Young, Olzomer, Ellen M, Beretta, Martina, Cantley, James, Nunemaker, Craig S, Hoehn, Kyle L, Byrne, Frances L. \"Investigating the Expression and Function of the Glucose Transporter GLUT6 in Obesity.\" International journal of molecular sciences, 2022, pp.  , doi: https://doi.org/10.3390/ijms23179798'], ['Mas, Caroline, Norwood, Suzanne J, Bugarcic, Andrea, Kinna, Genevieve, Leneva, Natalya, Kovtun, Oleksiy, Ghai, Rajesh, Ona Yanez, Lorena E, Davis, Jasmine L, Teasdale, Rohan D, Collins, Brett M. \"Structural basis for different phosphoinositide specificities of the PX domains of sorting nexins regulating G-protein signaling.\" The Journal of biological chemistry, 2014, pp. 28554-68, doi: https://doi.org/10.1074/jbc.M114.595959', 'Du, Yingshi, Zou, Yan, Yu, Weihua, Shi, Rui, Zhang, Man, Yang, Wenxiu, Duan, Jingxi, Deng, Yongtao, Wang, Xuefeng, Lü, Yang. \"Expression pattern of sorting Nexin 25 in temporal lobe epilepsy: a study on patients and pilocarpine-induced rats.\" Brain research, 2013, pp. 79-85, doi: https://doi.org/10.1016/j.brainres.2013.03.005', 'Zhang, Yulong, Chen, Rui, Dong, Yan, Zhu, Jiabin, Su, Kai, Liu, Jinsong, Xu, Jinxin. \"Structural Studies Reveal Unique Non-canonical Regulators of G Protein Signaling Homology (RH) Domains in Sorting Nexins.\" Journal of molecular biology, 2022, pp. 167823, doi: https://doi.org/10.1016/j.jmb.2022.167823'], [], [], ['Yang, Yongfeng, Yang, Chuanzhen, Li, Tingting, Yu, Shuyu, Gan, Tingting, Hu, Jiazhi, Cui, Jun, Zheng, Xiaofeng. \"The Deubiquitinase USP38 Promotes NHEJ Repair through Regulation of HDAC1 Activity and Regulates Cancer Cell Response to Genotoxic Insults.\" Cancer research, 2020, pp. 719-731, doi: https://doi.org/10.1158/0008-5472.CAN-19-2149', 'Liu, Wenbin, Zhang, Qi, Fang, Yuanyuan, Wang, Yanan. \"The deubiquitinase USP38 affects cellular functions through interacting with LSD1.\" Biological research, 2018, pp. 53, doi: https://doi.org/10.1186/s40659-018-0201-8', 'Zhang, Tao, Su, Fei, Wang, Bofang, Liu, Lixin, Lu, Yongbin, Su, Hongxin, Ling, Ruijiang, Yue, Peng, Dai, Huanyu, Yang, Tianning, Yang, Jingru, Chen, Rui, Wu, Ruiyue, Zhu, Kaili, Zhao, Da, Hou, Xiaoming. \"Ubiquitin specific peptidase 38 epigenetically regulates KLF transcription factor 5 to augment malignant progression of lung adenocarcinoma.\" Oncogene, 2024, pp. 1190-1202, doi: https://doi.org/10.1038/s41388-024-02985-7'], ['Yan, Xiaojie, Li, Yao, Wang, Guobin, Zhou, Zhili, Song, Guangyong, Feng, Qiqi, Zhao, Yueling, Mi, Wenyi, Ma, Zhenyi, Dong, Cheng. \"Molecular basis for recognition of Gly/N-degrons by CRL2<sup>ZYG11B</sup> and CRL2<sup>ZER1</sup>.\" Molecular cell, 2021, pp. 3262-3274.e3, doi: https://doi.org/10.1016/j.molcel.2021.06.010', 'Zhang, Bing, Li, Yao, Feng, Qiqi, Song, Lili, Dong, Cheng, Yan, Xiaojie. \"Structural insights into ORF10 recognition by ZYG11B.\" Biochemical and biophysical research communications, 2022, pp. 14-18, doi: https://doi.org/10.1016/j.bbrc.2022.05.069'], []]\n"
     ]
    }
   ],
   "source": [
    "## UPDATES 02/13/2023\n",
    "#iterate through the df and save df and dict\n",
    "\n",
    "res_df, res_dict = iter_dataframe(df, email, config, n=3, papers_query=20, verbose=False, return_paragraph_ref_data=False, id_col=nameCol, paragraph_col=LLM_analysisCol, runVersion=runVersion, save_path = toSaveFilePath)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "======================================="
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## use your own iterate dataframe loop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "LLM_analysisFilePath = './data/CH_omics_revamped_LLM_Enrichr_simVals_failure_refs_newThresh_DF.tsv'\n",
    "toSaveFilePath  = './data/CH_omics_revamped_LLM_Enrichr_simVals_failure_refs_newThresh_DF'\n",
    "jsonFilePath = 'jsonFiles/reference_checking_revision_test.json'\n",
    "with open(jsonFilePath) as json_file:\n",
    "    config = json.load(json_file)\n",
    "\n",
    "openai.api_key = os.environ[\"OPENAI_API_KEY\"]\n",
    "email = config['EMAIL']\n",
    "nameCol = 'GeneSetName'\n",
    "LLM_analysisCol = 'LLM Analysis'\n",
    "runVersion = 'additional'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "runOnlyExamples = True\n",
    "if runOnlyExamples:\n",
    "    examplesTORun = ['BRD-A00993607 Alprenolol hydrochloride MCF7 6.0 h 10.0 um',\n",
    "       'BRD-A13964793 -666 MCF7 6.0 h 10.0 um',\n",
    "       'BRD-A19633847 PERHEXILINE MALEATE MCF7 6.0 h 10.0 um',\n",
    "       'BRD-A31204924 -666 MCF7 6.0 h 10.0 um', 'Cluster2-126',\n",
    "       'Cluster2-140', 'Cluster2-147', 'Cluster2-169', 'Cluster2-183',\n",
    "       'Cluster2-191', 'Cluster2-200']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Read in the LLM analysis file\n",
    "df = pd.read_csv(LLM_analysisFilePath, sep='\\t', keep_default_na=False, na_values=['NaN'])\n",
    "df = df.replace({'None':None})\n",
    "df.set_index(nameCol, inplace=True)\n",
    "# # rename 'References' to 'referenced_analysis' \n",
    "# df = df.rename(columns={'References':'referenced_analysis'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0.1</th>\n",
       "      <th>...1</th>\n",
       "      <th>...2</th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Source</th>\n",
       "      <th>GeneSetID</th>\n",
       "      <th>GeneList</th>\n",
       "      <th>n_Genes</th>\n",
       "      <th>LLM Name</th>\n",
       "      <th>LLM Analysis</th>\n",
       "      <th>...</th>\n",
       "      <th>Term</th>\n",
       "      <th>GO term</th>\n",
       "      <th>GO ID</th>\n",
       "      <th>GO_term_genes</th>\n",
       "      <th>LLM_name_GO_term_sim</th>\n",
       "      <th>enrichr_JI</th>\n",
       "      <th>LLM_success_TF</th>\n",
       "      <th>enrichr_success_TF</th>\n",
       "      <th>referenced_analysis</th>\n",
       "      <th>enrichr_success_TF_0.1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GeneSetName</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BRD-A31204924 -666 MCF7 6.0 h 10.0 um</th>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "      <td>9762</td>\n",
       "      <td>1</td>\n",
       "      <td>L1000</td>\n",
       "      <td>BRD-A31204924_-666_MCF7_6.0_h_10.0_um</td>\n",
       "      <td>1060P11.3 ANPEP DPP4 FAM129A HNMT MKNK1 MUC5B ...</td>\n",
       "      <td>15</td>\n",
       "      <td>System of unrelated proteins</td>\n",
       "      <td>1. 1060P11.3 is a hypothetical protein with li...</td>\n",
       "      <td>...</td>\n",
       "      <td>Histamine Metabolic Process (GO:0001692)</td>\n",
       "      <td>Histamine Metabolic Process</td>\n",
       "      <td>GO:0001692</td>\n",
       "      <td>SLC22A3 PRG3 TRH HNMT SLC29A4</td>\n",
       "      <td>0.264298</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Unnamed: 0.1  ...1  ...2  Unnamed: 0  \\\n",
       "GeneSetName                                                                   \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um            85    85  9762           1   \n",
       "\n",
       "                                      Source  \\\n",
       "GeneSetName                                    \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um  L1000   \n",
       "\n",
       "                                                                   GeneSetID  \\\n",
       "GeneSetName                                                                    \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um  BRD-A31204924_-666_MCF7_6.0_h_10.0_um   \n",
       "\n",
       "                                                                                GeneList  \\\n",
       "GeneSetName                                                                                \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um  1060P11.3 ANPEP DPP4 FAM129A HNMT MKNK1 MUC5B ...   \n",
       "\n",
       "                                       n_Genes                      LLM Name  \\\n",
       "GeneSetName                                                                    \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um       15  System of unrelated proteins   \n",
       "\n",
       "                                                                            LLM Analysis  \\\n",
       "GeneSetName                                                                                \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um  1. 1060P11.3 is a hypothetical protein with li...   \n",
       "\n",
       "                                       ...  \\\n",
       "GeneSetName                            ...   \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um  ...   \n",
       "\n",
       "                                                                           Term  \\\n",
       "GeneSetName                                                                       \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um  Histamine Metabolic Process (GO:0001692)   \n",
       "\n",
       "                                                            GO term  \\\n",
       "GeneSetName                                                           \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um  Histamine Metabolic Process    \n",
       "\n",
       "                                            GO ID  \\\n",
       "GeneSetName                                         \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um  GO:0001692   \n",
       "\n",
       "                                                       GO_term_genes  \\\n",
       "GeneSetName                                                            \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um  SLC22A3 PRG3 TRH HNMT SLC29A4   \n",
       "\n",
       "                                       LLM_name_GO_term_sim enrichr_JI  \\\n",
       "GeneSetName                                                              \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um              0.264298   0.052632   \n",
       "\n",
       "                                      LLM_success_TF enrichr_success_TF  \\\n",
       "GeneSetName                                                               \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um          False              False   \n",
       "\n",
       "                                      referenced_analysis  \\\n",
       "GeneSetName                                                 \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um                       \n",
       "\n",
       "                                      enrichr_success_TF_0.1  \n",
       "GeneSetName                                                   \n",
       "BRD-A31204924 -666 MCF7 6.0 h 10.0 um                  False  \n",
       "\n",
       "[1 rows x 26 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['referenced_analysis'] == '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if runVersion == 'initial':\n",
    "    df['referenced_analysis'] = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=========================================\n",
      "=========================================\n",
      "=========================================\n",
      "['dataframe row', 'BRD-A31204924 -666 MCF7 6.0 h 10.0 um']\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "1. 1060P11.3 is a hypothetical protein with limited information available regarding its function. Without clear evidence of its biological role, it cannot be easily integrated into a specific process with the other proteins listed.\n",
      "===========================================================================\n",
      "177\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "1. 1060P11.3 is a hypothetical protein with limited information available regarding its function. Without clear evidence of its biological role, it cannot be easily integrated into a specific process with the other proteins listed.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "Unknown\n",
      "291\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "1. 1060P11.3 is a hypothetical protein with limited information available regarding its function. Without clear evidence of its biological role, it cannot be easily integrated into a specific process with the other proteins listed.\n",
      "    \n",
      "Result:\n",
      "hypothetical protein,protein function,biological role\n",
      "No keyword generated skip referencing\n",
      "In paragraph 1, 0 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "2. ANPEP, also known as CD13, is a membrane-bound aminopeptidase involved in the metabolism of regulatory peptides by diverse cell types, and plays a role in angiogenesis and tumor invasion.\n",
      "===========================================================================\n",
      "180\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "2. ANPEP, also known as CD13, is a membrane-bound aminopeptidase involved in the metabolism of regulatory peptides by diverse cell types, and plays a role in angiogenesis and tumor invasion.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "ANPEP\n",
      "292\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "2. ANPEP, also known as CD13, is a membrane-bound aminopeptidase involved in the metabolism of regulatory peptides by diverse cell types, and plays a role in angiogenesis and tumor invasion.\n",
      "    \n",
      "Result:\n",
      "aminopeptidase,angiogenesis,tumor invasion\n",
      "PubMed Keywords:  (\"ANPEP\") AND (\"aminopeptidase\" OR \"angiogenesis\" OR \"tumor invasion\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "158\n",
      "157\n",
      "158\n",
      "Prognostic significance of aberrantly silenced ANPEP expression in prostate cancer. ['ANPEP'] 32\n",
      "161\n",
      "183\n",
      "159\n",
      "161\n",
      "159\n",
      "155\n",
      "168\n",
      "187\n",
      "LncRNA TUG1 silencing enhances proliferation and migration of ox-LDL-treated human umbilical vein endothelial cells and promotes atherosclerotic vascular injury repairing via the Runx2/ANPEP axis. ['ANPEP'] 5\n",
      "162\n",
      "Aminopeptidase N expression in the endometrium could affect endometrial receptivity. ['ANPEP'] 4\n",
      "188\n",
      "Soluble ANPEP Released From Human Astrocytes as a Positive Regulator of Microglial Activation and Neuroinflammation: Brain Renin-Angiotensin System in Astrocyte-Microglia Crosstalk. ['ANPEP'] 3\n",
      "162\n",
      "The number of title matching paper: 4\n",
      "465\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "2. ANPEP, also known as CD13, is a membrane-bound aminopeptidase involved in the metabolism of regulatory peptides by diverse cell types, and plays a role in angiogenesis and tumor invasion.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Novel biomarkers for prostate cancer (PC) are urgently needed. This study investigates the expression, epigenetic regulation, and prognostic potential of ANPEP in PC.\n",
      "Aminopeptidase N (APN; encoded by ANPEP) expression was analysed by immunohistochemistry using tissue microarrays representing 267 radical prostatectomy (RP) and 111 conservatively treated (CT) PC patients. Clinical end points were recurrence-free survival (RFS) and cancer-specific survival (CSS), respectively. The ANPEP promoter methylation levels were determined by bisulphite sequencing or MethyLight analysis in 278 nonmalignant and PC tissue samples, and in cell lines.\n",
      "The APN expression was significantly downregulated in PC compared with nonmalignant prostate tissue samples. Aberrant promoter hypermethylation was frequently observed in PC tissue samples, and 5-aza-2'-deoxycytidine induced ANPEP expression in three hypermethylated prostate cell lines, suggesting epigenetic silencing. Negative APN immunoreactivity was significantly associated with short RFS and short CSS in the RP and CT cohort, respectively, independently of routine clinicopathological predictors. Combining APN with a known angiogenesis marker (vascular endothelial growth factor or microvessel density) improved risk prediction significantly in both cohorts.\n",
      "Our results suggest negative APN immunoreactivity as a new independent adverse prognostic factor for patients with clinically localised PC and, furthermore, that epigenetic mechanisms are involved in silencing of ANPEP in PC.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  ANPEP\n",
      "Reduced expression of alanyl aminopeptidase is a robust biomarker of non-familial adenomatous polyposis and non-hereditary nonpolyposis colorectal cancer syndrome early-onset colorectal cancer.\n",
      "515\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "2. ANPEP, also known as CD13, is a membrane-bound aminopeptidase involved in the metabolism of regulatory peptides by diverse cell types, and plays a role in angiogenesis and tumor invasion.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "The role of vascular endothelial cell injury in the course of atherosclerosis (AS) has attracted increasing attention. Long non-coding RNAs (LncRNAs) are demonstrated to be the biomarker for the diagnosis of AS. This study investigated the mechanism of lncRNA taurine upregulated gene 1 (TUG1) in AS. Microarray data of AS obtained from GEO database showed that lncRNA TUG1 was differentially expressed in AS samples. TUG1 expression was upregulated in ox-LDL-treated human umbilical vein endothelial cells (HUVECs). Oxidized low density lipoprotein (ox-LDL)-treated HUVECs were then transfected with sh-TUG1. TUG1 silencing promoted proliferation and migration of ox-LDL-treated HUVECs. TUG1 bound to Runt-related transcription factor 2 (Runx2). Runx2 silencing promoted proliferation and migration of HUVECs. The downstream genes of Runx2 were predicted by hTFtarget database. The binding site of Runx2 and Aminopeptidase N (ANPEP) was determined. Runx2 silencing reversed the repression effect of overexpressing ANPEP on cell proliferation and migration. TUG1 silencing inhibited ANPEP expression via Runx2 to promote HUVEC proliferation and migration. A mouse model of AS was established. The area of atherosclerotic lesions of mouse aorta was detected, and vascular re-endothelialization was evaluated. TUG1 silencing promoted vascular injury repairing and inhibited AS in vivo. In conclusion, TUG1 silencing enhanced proliferation and migration of ox-LDL-treated HUVECs and promoted vascular injury repairing in vivo via the Runx2/ANPEP axis.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  No.\n",
      "467\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "2. ANPEP, also known as CD13, is a membrane-bound aminopeptidase involved in the metabolism of regulatory peptides by diverse cell types, and plays a role in angiogenesis and tumor invasion.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Aminopeptidase N (ANPEP) is a membrane-bound zinc-dependent peptidase. Although it is widely believed that ANPEP acts as an important angiogenesis regulatory factor, there are few studies about its function in the female reproductive system. In our previous research, we applied Isobaric Tags for Relative and Absolute Quantitation (iTRAQ) to analyze the influence of different controlled superovulation treatments for Assisted Reproductive Technology, In Vitro Fertilization and Embryo Transfer (IVF-ET)) patients from a global proteomic perspective to search for potential biomarkers associated with endometrium receptivity and embryo implantation. ANPEP is one of the proteins that demonstrated differential expression between different treatment groups and may be closely associated with endometrial receptivity. In this study, we assessed the expression of ANPEP in the endometrium of mice at different ages and found it to be highest in the mature period. We also detected ANPEP expression in the endometrium of IVF-ET patients in the proliferative, preimplantation and implantation stages, and the highest expression level of ANPEP was found in the last group. Human primary endometrial stromal cells were infected with an adenovirus expression vector containing the ANPEP gene and a green fluorescent protein (GFP) fusion protein; the mRNA levels of HOXA-10, LIF, and integrin β3 were found to be increased. Therefore, we conclude that ANPEP could be involved in the regulation of endometrial receptivity.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "570\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "2. ANPEP, also known as CD13, is a membrane-bound aminopeptidase involved in the metabolism of regulatory peptides by diverse cell types, and plays a role in angiogenesis and tumor invasion.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Astrocytes are major supportive glia and immune modulators in the brain; they are highly secretory in nature and interact with other cell types via their secreted proteomes. To understand how astrocytes communicate during neuroinflammation, we profiled the secretome of human astrocytes following stimulation with proinflammatory factors. A total of 149 proteins were significantly upregulated in stimulated astrocytes, and a bioinformatics analysis of the astrocyte secretome revealed that the brain renin-angiotensin system (RAS) is an important mechanism of astrocyte communication. We observed that the levels of soluble form of aminopeptidase N (sANPEP), an RAS component that converts angiotensin (Ang) III to Ang IV in a neuroinflammatory milieu, significantly increased in the astrocyte secretome. To elucidate the role of sANPEP and Ang IV in neuroinflammation, we first evaluated the expression of Ang IV receptors in human glial cells because Ang IV mediates biological effects through its receptors. The expression of angiotensin type 1 receptor was considerably upregulated in activated human microglial cells but not in human astrocytes. Moreover, interleukin-1β release from human microglial cells was synergistically increased by cotreatment with sANPEP and its substrate, Ang III, suggesting the proinflammatory action of Ang IV generated by sANPEP. In a mouse neuroinflammation model, brain microglial activation and proinflammatory cytokine expression levels were increased by intracerebroventricular injection of sANPEP and attenuated by an enzymatic inhibitor and neutralizing antibody against sANPEP. Collectively, our results indicate that astrocytic sANPEP-induced increase in Ang IV exacerbates neuroinflammation by interacting with microglial proinflammatory receptor angiotensin type 1 receptor, highlighting an important role of indirect crosstalk between astrocytes and microglia through the brain RAS in neuroinflammation.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 2, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "3. DPP4, or dipeptidyl peptidase-4, is an enzyme that cleaves X-proline dipeptides from the N-terminus of polypeptides. It is involved in glucose metabolism by inactivating incretin hormones and has a role in immune regulation.\n",
      "===========================================================================\n",
      "198\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "3. DPP4, or dipeptidyl peptidase-4, is an enzyme that cleaves X-proline dipeptides from the N-terminus of polypeptides. It is involved in glucose metabolism by inactivating incretin hormones and has a role in immune regulation.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "DPP4\n",
      "316\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "3. DPP4, or dipeptidyl peptidase-4, is an enzyme that cleaves X-proline dipeptides from the N-terminus of polypeptides. It is involved in glucose metabolism by inactivating incretin hormones and has a role in immune regulation.\n",
      "    \n",
      "Result:\n",
      "glucose metabolism,dipeptidyl peptidase-4,immune regulation\n",
      "PubMed Keywords:  (\"DPP4\") AND (\"glucose metabolism\" OR \"dipeptidyl peptidase-4\" OR \"immune regulation\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "179\n",
      "Dipeptidyl peptidase 4 inhibitors and their potential immune modulatory functions. ['DPP4'] 74\n",
      "182\n",
      "Potential disease biomarkers: dipeptidyl peptidase 4 and fibroblast activation protein. ['DPP4'] 23\n",
      "174\n",
      "DPP4 Activity, Hyperinsulinemia, and Atherosclerosis. ['DPP4'] 12\n",
      "178\n",
      "179\n",
      "Emerging Role of Dipeptidyl Peptidase-4 in Autoimmune Disease. ['DPP4'] 10\n",
      "174\n",
      "The regulatory role of DPP4 in atherosclerotic disease. ['DPP4'] 9\n",
      "179\n",
      "Microbial-host-isozyme analyses reveal microbial DPP4 as a potential antidiabetic target. ['DPP4'] 7\n",
      "176\n",
      "Context-dependent effects of dipeptidyl peptidase 4 inhibitors. ['DPP4'] 7\n",
      "178\n",
      "An expanded role for dipeptidyl peptidase 4 in cell regulation. ['DPP4'] 7\n",
      "168\n",
      "The number of title matching paper: 8\n",
      "419\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "3. DPP4, or dipeptidyl peptidase-4, is an enzyme that cleaves X-proline dipeptides from the N-terminus of polypeptides. It is involved in glucose metabolism by inactivating incretin hormones and has a role in immune regulation.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Dipeptidyl peptidase 4 (DPP4) inhibitors (DPP4is) are oral anti-diabetic drugs (OADs) for the treatment of type 2 diabetes mellitus (T2DM) through inhibiting the degradation of incretin peptides. Numerous investigations have been focused on the effects of DPP4is on glucose homeostasis. However, there are limited evidences demonstrating their Potential modulatory functions in the immune system. DPP4, originally known as the lymphocyte cell surface protein CD26, is widely expressed in many types of immune cells including CD4(+) and CD8(+) T cells, B cells, NK cells, dendritic cells, and macrophages; and regulate the functions of these cells. In addition, DPP4 is capable of modulating plenty of cytokines, chemokines and peptide hormones. Accordingly, DPP4/CD26 is speculated to be involved in various immune/inflammatory diseases and DPP4is may become a new drug class applied in these diseases. This review focuses on the regulatory effects of DPP4is on immune functions and their possible underlying mechanisms. Further clinical studies will be necessitated to fully evaluate the administration of DPP4is in diabetic patients with or without immune diseases.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  DPP4\n",
      "Evogliptin for the treatment option for type 2 diabetes: an update of the literature.\n",
      "323\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "3. DPP4, or dipeptidyl peptidase-4, is an enzyme that cleaves X-proline dipeptides from the N-terminus of polypeptides. It is involved in glucose metabolism by inactivating incretin hormones and has a role in immune regulation.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "The importance of the dipeptidyl peptidase 4 (DPP4) gene family in regulating critical biochemical pathways continues to emerge. The two most well-studied members of the family, DPP4 and fibroblast activation protein (FAP), have been investigated both as therapeutic targets for disease and as diagnostic biomarkers. The interest in DPP4 and FAP as potential disease biomarkers has been driven primarily by observations of altered expression profiles in inflammatory diseases and cancer. Furthermore, the stability and persistence of soluble DPP4 and FAP in the serum make them attractive candidate serology markers. This review summarises investigations into DPP4 and FAP as biomarkers of autoimmune disease, gut inflammation, psychosomatic disorders and malignancy and discusses their potential likelihood as clinically useful tools.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "512\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "3. DPP4, or dipeptidyl peptidase-4, is an enzyme that cleaves X-proline dipeptides from the N-terminus of polypeptides. It is involved in glucose metabolism by inactivating incretin hormones and has a role in immune regulation.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Obesity and type 2 diabetes are associated with chronic hyperinsulinemia, elevated plasma levels of dipeptidyl peptidase-4 (DPP4), and a pro-atherosclerotic milieu.\n",
      "PubMed search of the term \"insulin and atherosclerosis,\" \"hyperinsulinemia,\" \"atherosclerosis,\" or \"cardiovascular outcomes\" cross-referenced with \"DPP4.\" Relevant research and review articles were reviewed.\n",
      "Hyperinsulinemia in the setting of insulin resistance promotes vascular inflammation, vascular smooth muscle cell growth, pathological cholesterol profile, hypertension, and recruitment of immune cells to the endothelium, all contributing to atherosclerosis. DPP4 has pleiotropic functions and its activity is elevated in obese humans. DPP4 mirrors hyperinsulinemia's atherogenic actions in the insulin resistant state, and genetic deletion of DPP4 protects rodents from developing insulin resistance and improves cardiovascular outcomes. DPP4 inhibition in pro-atherosclerotic preclinical models results in reduced inflammation and oxidative stress, improved endothelial function, and decreased atherosclerosis. Increased incretin levels may have contributed to but do not completely account for these benefits. Small clinical studies with DPP4 inhibitors demonstrate reduced carotid intimal thickening, improved endothelial function, and reduced arterial stiffness. To date, this has not been translated to cardiovascular risk reduction for individuals with type 2 diabetes with prior or exaggerated risk of cardiovascular disease.\n",
      "DPP4 may represent a key link between central obesity, insulin resistance, and atherosclerosis. The gaps in knowledge in DPP4 function and discrepancy in cardiovascular outcomes observed in preclinical and large-scale randomized controlled studies with DPP4 inhibitors warrant additional research.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 3, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "4. FAM129A, also known as Niban, is a protein that may be involved in cellular stress response pathways and has been implicated in cancer progression.\n",
      "===========================================================================\n",
      "169\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "4. FAM129A, also known as Niban, is a protein that may be involved in cellular stress response pathways and has been implicated in cancer progression.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "FAM129A\n",
      "279\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "4. FAM129A, also known as Niban, is a protein that may be involved in cellular stress response pathways and has been implicated in cancer progression.\n",
      "    \n",
      "Result:\n",
      "cellular stress response,cancer progression,protein function\n",
      "PubMed Keywords:  (\"NIBAN1\") AND (\"cellular stress response\" OR \"cancer progression\" OR \"protein function\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "3 references are queried\n",
      "The number of title matching paper: 0\n",
      "In paragraph 4, 0 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "5. HNMT, histamine N-methyltransferase, is involved in the metabolism of histamine, and thus plays a role in the regulation of immune responses and gastric acid secretion.\n",
      "===========================================================================\n",
      "173\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "5. HNMT, histamine N-methyltransferase, is involved in the metabolism of histamine, and thus plays a role in the regulation of immune responses and gastric acid secretion.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "HNMT\n",
      "287\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "5. HNMT, histamine N-methyltransferase, is involved in the metabolism of histamine, and thus plays a role in the regulation of immune responses and gastric acid secretion.\n",
      "    \n",
      "Result:\n",
      "histamine metabolism,immune regulation,gastric acid secretion\n",
      "PubMed Keywords:  (\"HNMT\") AND (\"histamine metabolism\" OR \"immune regulation\" OR \"gastric acid secretion\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "148\n",
      "Altered expression of histamine signaling genes in autism spectrum disorder. ['HNMT'] 59\n",
      "156\n",
      "Human histamine N-methyltransferase pharmacogenetics: common genetic polymorphisms that alter activity. ['HNMT'] 31\n",
      "162\n",
      "Recent advances in molecular pharmacology of the histamine systems: organic cation transporters as a histamine transporter and histamine metabolism. ['HNMT'] 27\n",
      "152\n",
      "Structural basis for inhibition of histamine N-methyltransferase by diverse drugs. ['HNMT'] 26\n",
      "140\n",
      "Histamine pharmacogenomics. ['HNMT'] 20\n",
      "151\n",
      "Histamine <i>N</i>-Methyltransferase in the Brain. ['HNMT'] 15\n",
      "157\n",
      "Decreased histamine catabolism in the colonic mucosa of patients with colonic adenoma. ['HNMT'] 13\n",
      "166\n",
      "167\n",
      "Association of diamine oxidase and histamine N-methyltransferase polymorphisms with presence of migraine in a group of Mexican mothers of children with allergies. ['HNMT'] 8\n",
      "152\n",
      "Actions of tacrine and galanthamine on histamine-N-methyltransferase. ['HNMT'] 6\n",
      "153\n",
      "160\n",
      "Impact of the clinically approved Petasites hybridus extract Ze 339 on intestinal mechanisms involved in the handling of histamine. ['HNMT'] 4\n",
      "165\n",
      "Inhibition of Astrocytic Histamine <i>N</i>-Methyltransferase as a Possible Target for the Treatment of Alzheimer's Disease. ['HNMT'] 2\n",
      "158\n",
      "Mapping of the binding sites of human histamine N-methyltransferase (HNMT) monoclonal antibodies. ['HNMT'] 2\n",
      "146\n",
      "154\n",
      "Characterization of a new mRNA species from the human histamine N-methyltransferase gene. ['HNMT'] 0\n",
      "The number of title matching paper: 13\n",
      "468\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "5. HNMT, histamine N-methyltransferase, is involved in the metabolism of histamine, and thus plays a role in the regulation of immune responses and gastric acid secretion.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "The histaminergic system (HS) has a critical role in cognition, sleep and other behaviors. Although not well studied in autism spectrum disorder (ASD), the HS is implicated in many neurological disorders, some of which share comorbidity with ASD, including Tourette syndrome (TS). Preliminary studies suggest that antagonism of histamine receptors 1-3 reduces symptoms and specific behaviors in ASD patients and relevant animal models. In addition, the HS mediates neuroinflammation, which may be heightened in ASD. Together, this suggests that the HS may also be altered in ASD. Using RNA sequencing (RNA-seq), we investigated genome-wide expression, as well as a focused gene set analysis of key HS genes (HDC, HNMT, HRH1, HRH2, HRH3 and HRH4) in postmortem dorsolateral prefrontal cortex (DLPFC) initially in 13 subjects with ASD and 39 matched controls. At the genome level, eight transcripts were differentially expressed (false discovery rate <0.05), six of which were small nucleolar RNAs (snoRNAs). There was no significant diagnosis effect on any of the individual HS genes but expression of the gene set of HNMT, HRH1, HRH2 and HRH3 was significantly altered. Curated HS gene sets were also significantly differentially expressed. Differential expression analysis of these gene sets in an independent RNA-seq ASD data set from DLPFC of 47 additional subjects confirmed these findings. Understanding the physiological relevance of an altered HS may suggest new therapeutic options for the treatment of ASD.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  HNMT\n",
      "Genetic variation within the histamine pathway among patients with asthma--a pilot study.\n",
      "494\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "5. HNMT, histamine N-methyltransferase, is involved in the metabolism of histamine, and thus plays a role in the regulation of immune responses and gastric acid secretion.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Histamine N-methyltransferase (HNMT) catalyzes a major pathway in histamine metabolism. Levels of HNMT activity in humans are regulated by inheritance. We set out to study the molecular basis for this genetic regulation. Northern blot analysis showed that HNMT is highly expressed in the kidney, so we determined levels of enzyme activity and thermal stability in 127 human renal biopsy samples. DNA was isolated from 12 kidney samples with widely different HNMT phenotypes, and exons of the HNMT gene were amplified with the polymerase chain reaction. In these 12 samples, we observed a C314T transition that resulted in a Thr105Ile change in encoded amino acid, as well as an A939G transition within the 3'-untranslated region. All remaining renal biopsy samples then were genotyped for these two variant sequences. Frequencies of the alleles encoding Thr105 and Ile105 in the 114 samples studied were 0.90 and 0.10, respectively, whereas frequencies for the nucleotide A939 and G alleles were 0.79 and 0.21, respectively. Kidney samples with the allele encoding Ile105 had significantly lower levels of HNMT activity and thermal stability than did those with the allele that encoded Thr105. These observations were confirmed by transient expression in COS-1 cells of constructs that contained all four alleles for these two polymorphisms. COS-1 cells transfected with the Ile105 allele had significantly lower HNMT activity and immunoreactive HNMT protein than did those transfected with the Thr105 allele. These observations will make it possible to test the hypothesis that genetic polymorphisms for HNMT may play a role in the pathophysiology of human disease.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "614\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "5. HNMT, histamine N-methyltransferase, is involved in the metabolism of histamine, and thus plays a role in the regulation of immune responses and gastric acid secretion.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Histamine is inactivated by the histamine-metabolizing enzyme histamine N-methyltransferase (HNMT) in bronchus, kidney, and the central nervous system. HNMT seems to be localized in the cytoplasm, but histamine is unable to easily enter the intracellular space. Therefore, two hypotheses can be elicited: one is the plasma membrane hypothesis that HNMT can be translocated to the plasma membrane and function at the cell surface under growth factor stimulation and the other is the transporter hypothesis that organic cation transporter (OCT)-2 and -3 can function as a histamine transporter as well. To investigate the involvement of OCT2, HEK293 cells stably double transfected with C-terminal hemagglutinin (HA)-tagged HNMT cDNA and/or C-terminal myc-tagged rat OCT2 were prepared for analysis of HNMT activity associated with OCT2 function. After 60-min incubation of these cells with PBS including HA (100 microM), N(tau)-methylhistamine (MHA) concentration of the supernatants was determined by the HPLC-fluorometry method. MHA from cells with HNMT plus OCT-2 was produced at about 3-fold higher level than that from cells with HNMT alone, suggesting that OCT-2 could function as a histamine transporter as well and that HNMT function could partly depend on OCT-2 transporter activity. Using OCT-3 knockout (OCT-3-/-) mice, histamine content and survival rates were investigated in lipopolysaccharide (LPS)-induced endotoxemia model. Without LPS stimulation, histamine content was compared between OCT-3-/- and wild mice. Histamine content in the spleen of OCT-3-/- mice was higher than that f wild mice. With LPS stimulation, the survival rate of OCT-3-/- mice was significantly decreased 12 h after LPS (20 mg/kg) stimulation, suggesting that before immunological stimulation, a higher content of histamine in spleen could stimulate histamine receptors in mast cells, macrophages, dendritic cells, as well as T lymphocytes and explaining the decreased survival rate in OCT-3-/- mice possibly due to the functional changes of immunological cells.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 5, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "6. MKNK1, MAP kinase-interacting serine/threonine-protein kinase 1, is involved in the regulation of mRNA translation under stress conditions and has been implicated in cancer cell proliferation.\n",
      "===========================================================================\n",
      "178\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "6. MKNK1, MAP kinase-interacting serine/threonine-protein kinase 1, is involved in the regulation of mRNA translation under stress conditions and has been implicated in cancer cell proliferation.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "MKNK1\n",
      "288\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "6. MKNK1, MAP kinase-interacting serine/threonine-protein kinase 1, is involved in the regulation of mRNA translation under stress conditions and has been implicated in cancer cell proliferation.\n",
      "    \n",
      "Result:\n",
      "mRNA translation,stress response,cancer cell proliferation\n",
      "PubMed Keywords:  (\"MKNK1\") AND (\"mRNA translation\" OR \"stress response\" OR \"cancer cell proliferation\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "170\n",
      "MNK Inhibition Disrupts Mesenchymal Glioma Stem Cells and Prolongs Survival in a Mouse Model of Glioblastoma. ['MKNK1'] 24\n",
      "The number of title matching paper: 1\n",
      "507\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "6. MKNK1, MAP kinase-interacting serine/threonine-protein kinase 1, is involved in the regulation of mRNA translation under stress conditions and has been implicated in cancer cell proliferation.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Glioblastoma multiforme remains the deadliest malignant brain tumor, with glioma stem cells (GSC) contributing to treatment resistance and tumor recurrence. We have identified MAPK-interacting kinases (MNK) as potential targets for the GSC population in glioblastoma multiforme. Isoform-level subtyping using The Cancer Genome Atlas revealed that both MNK genes (MKNK1 and MKNK2) are upregulated in mesenchymal glioblastoma multiforme as compared with other subtypes. Expression of MKNK1 is associated with increased glioma grade and correlated with the mesenchymal GSC marker, CD44, and coexpression of MKNK1 and CD44 predicts poor survival in glioblastoma multiforme. In established and patient-derived cell lines, pharmacologic MNK inhibition using LY2801653 (merestinib) inhibited phosphorylation of the eukaryotic translation initiation factor 4E, a crucial effector for MNK-induced mRNA translation in cancer cells and a marker of transformation. Importantly, merestinib inhibited growth of GSCs grown as neurospheres as determined by extreme limiting dilution analysis. When the effects of merestinib were assessed in vivo using an intracranial xenograft mouse model, improved overall survival was observed in merestinib-treated mice. Taken together, these data provide strong preclinical evidence that pharmacologic MNK inhibition targets mesenchymal glioblastoma multiforme and its GSC population.\n",
      "These findings raise the possibility of MNK inhibition as a viable therapeutic approach to target the mesenchymal subtype of glioblastoma multiforme. Mol Cancer Res; 14(10); 984-93. ©2016 AACR.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  MKNK1\n",
      "Design, synthesis and biological evaluation of pyridone-aminal derivatives as MNK1/2 inhibitors.\n",
      "In paragraph 6, 1 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "7. MUC5B is a gel-forming mucin that plays a crucial role in the protection of the mucous membranes by forming a protective barrier and is involved in mucociliary clearance in the respiratory system.\n",
      "===========================================================================\n",
      "180\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "7. MUC5B is a gel-forming mucin that plays a crucial role in the protection of the mucous membranes by forming a protective barrier and is involved in mucociliary clearance in the respiratory system.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "MUC5B\n",
      "292\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "7. MUC5B is a gel-forming mucin that plays a crucial role in the protection of the mucous membranes by forming a protective barrier and is involved in mucociliary clearance in the respiratory system.\n",
      "    \n",
      "Result:\n",
      "mucin,mucociliary clearance,respiratory system\n",
      "PubMed Keywords:  (\"MUC5B\") AND (\"mucin\" OR \"mucociliary clearance\" OR \"respiratory system\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "153\n",
      "Mucins, Mucus, and Goblet Cells. ['MUC5B'] 135\n",
      "163\n",
      "Localization of Secretory Mucins MUC5AC and MUC5B in Normal/Healthy Human Airways. ['MUC5B'] 118\n",
      "151\n",
      "Mucins: the frontline defence of the lung. ['MUC5B'] 81\n",
      "150\n",
      "Biomarkers in idiopathic pulmonary fibrosis. ['MUC5B'] 56\n",
      "175\n",
      "Airway mucin MUC5AC and MUC5B concentrations and the initiation and progression of chronic obstructive pulmonary disease: an analysis of the SPIROMICS cohort. ['MUC5B'] 52\n",
      "166\n",
      "Defensive Properties of Mucin Glycoproteins during Respiratory Infections-Relevance for SARS-CoV-2. ['MUC5B'] 41\n",
      "148\n",
      "158\n",
      "Prevalence and Mechanisms of Mucus Accumulation in COVID-19 Lung Disease. ['MUC5B'] 18\n",
      "159\n",
      "Airway mucins promote immunopathology in virus-exacerbated chronic obstructive pulmonary disease. ['MUC5B'] 17\n",
      "158\n",
      "Assembly, Release, and Transport of Airway Mucins in Pigs and Humans. ['MUC5B'] 13\n",
      "156\n",
      "Intracellular Processing of Human Secreted Polymeric Airway Mucins. ['MUC5B'] 11\n",
      "167\n",
      "MUC5B mobilizes and MUC5AC spatially aligns mucociliary transport on human airway epithelium. ['MUC5B'] 8\n",
      "157\n",
      "151\n",
      "Mucins and CFTR: Their Close Relationship. ['MUC5B'] 5\n",
      "168\n",
      "Mucins MUC5AC and MUC5B Are Variably Packaged in the Same and in Separate Secretory Granules. ['MUC5B'] 5\n",
      "154\n",
      "Dysregulated Cell-Cell Communication Characterizes Pulmonary Fibrosis. ['MUC5B'] 2\n",
      "155\n",
      "The number of title matching paper: 14\n",
      "517\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "7. MUC5B is a gel-forming mucin that plays a crucial role in the protection of the mucous membranes by forming a protective barrier and is involved in mucociliary clearance in the respiratory system.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "The respiratory epithelium is lined by mucus, a gel consisting of water, ions, proteins, and macromolecules. The major macromolecular components of mucus are the mucin glycoproteins, which are critical for local defense of the airway. There are three classes of mucins in the airways: those that are secreted but do not polymerize (MUC7), those that are secreted and polymerize to form gels (MUC5AC, MUC5B), and those that have transmembrane domains and are cell surface associated (MUC1, MUC4, MUC16, MUC20). The mucins are regulated at the transcriptional, posttranscriptional, and epigenetic levels, and posttranslational modifications play an important role in mucin binding and clearance of microbes and pollutants. The development of mice deficient in specific mucins, and the cystic fibrosis pig, has greatly advanced our understanding of the role of mucins as innate immune mediators and how mucins and mucus contribute to lung disease. These observations suggest new strategies to ameliorate mucus obstruction by targeting mucociliary clearance and mucin hyperconcentration. Furthermore, a polymorphism in the promoter of MUC5B is strongly associated with risk of developing pulmonary fibrosis, supporting a novel function for MUC5B to influence interstitial lung disease. Exciting new data support the concept not only that mucins and mucus are important for lung homeostasis and protection from environmental threats but also that goblet cells play an important role as regulators of innate immune function. These insights into the innate immune properties of mucins and goblet cells support a shift from the current paradigm of repressing increased mucin expression to targeting regulation of specific mucins and the abnormal airway milieu.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  MUC5B\n",
      "Expression and Clinical Significance of Mucin Gene in Chronic Rhinosinusitis.\n",
      "544\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "7. MUC5B is a gel-forming mucin that plays a crucial role in the protection of the mucous membranes by forming a protective barrier and is involved in mucociliary clearance in the respiratory system.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "MUC5AC and MUC5B are the predominant gel-forming mucins in the mucus layer of human airways. Each mucin has distinct functions and site-specific expression. However, the regional distribution of expression and cell types that secrete each mucin in normal/healthy human airways are not fully understood.\n",
      "To characterize the regional distribution of MUC5B and MUC5AC in normal/healthy human airways and assess which cell types produce these mucins, referenced to the club cell secretory protein (CCSP).\n",
      "Multiple airway regions from 16 nonsmoker lungs without a history of lung disease were studied. MUC5AC, MUC5B, and CCSP expression/colocalization were assessed by RNA in situ hybridization and immunohistochemistry in five lungs with histologically healthy airways. Droplet digital PCR and cell cultures were performed for absolute quantification of MUC5AC/5B ratios and protein secretion, respectively.\n",
      "Submucosal glands expressed MUC5B, but not MUC5AC. However, MUC5B was also extensively expressed in superficial epithelia throughout the airways except for the terminal bronchioles. Morphometric calculations revealed that the distal airway superficial epithelium was the predominant site for MUC5B expression, whereas MUC5AC expression was concentrated in proximal, cartilaginous airways. RNA in situ hybridization revealed MUC5AC and MUC5B were colocalized with CCSP-positive secretory cells in proximal superficial epithelia, whereas MUC5B and CCSP-copositive cells dominated distal regions.\n",
      "In normal/healthy human airways, MUC5B is the dominant secretory mucin in the superficial epithelium and glands, with distal airways being a major site of expression. MUC5B and MUC5AC expression is a property of CCSP-positive secretory cells in superficial airway epithelia.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "379\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "7. MUC5B is a gel-forming mucin that plays a crucial role in the protection of the mucous membranes by forming a protective barrier and is involved in mucociliary clearance in the respiratory system.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Mucus plays a vital role in protecting the lungs from environmental factors, but conversely, in muco-obstructive airway disease, mucus becomes pathologic. In its protective role, mucus entraps microbes and particles removing them from the lungs via the co-ordinated beating of motile cilia. This mechanism of lung defence is reliant upon a flowing mucus gel, and the major macromolecular components that determine the rheological properties of mucus are the polymeric mucins, MUC5AC and MUC5B. These large O-linked glycoproteins have direct roles in maintaining lung homeostasis. MUC5B is essential for interaction with the ciliary clearance system and MUC5AC is up-regulated in response to allergic inflammatory challenge. Mucus with abnormal biophysical properties is a feature of muco-obstructive respiratory disease and can result from many different mechanisms including alterations in mucin polymer assembly, mucin concentration and the macromolecular form in mucus, as well as changes in airway surface hydration, pH and ion composition. The abnormal mucus results in defective lung protection via compromised ciliary clearance, leading to infection and inflammation.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 7, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "8. MYBPC1, myosin-binding protein C, slow-type, is a structural component of the muscle sarcomere that contributes to the regulation of muscle contraction.\n",
      "===========================================================================\n",
      "172\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "8. MYBPC1, myosin-binding protein C, slow-type, is a structural component of the muscle sarcomere that contributes to the regulation of muscle contraction.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "MYBPC1\n",
      "284\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "8. MYBPC1, myosin-binding protein C, slow-type, is a structural component of the muscle sarcomere that contributes to the regulation of muscle contraction.\n",
      "    \n",
      "Result:\n",
      "muscle contraction,MYBPC1,sarcomere\n",
      "PubMed Keywords:  (\"MYBPC1\") AND (\"muscle contraction\" OR \"MYBPC1\" OR \"sarcomere\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "159\n",
      "Myosin binding protein C1: a novel gene for autosomal dominant distal arthrogryposis type 1. ['MYBPC1'] 56\n",
      "164\n",
      "Slow skeletal muscle myosin-binding protein-C (MyBPC1) mediates recruitment of muscle-type creatine kinase (CK) to myosin. ['MYBPC1'] 24\n",
      "149\n",
      "Post-translational control of cardiac hemodynamics through myosin binding protein C. ['MYBPC1'] 15\n",
      "161\n",
      "Heterozygous variants in MYBPC1 are associated with an expanded neuromuscular phenotype beyond arthrogryposis. ['MYBPC1'] 9\n",
      "163\n",
      "Expanding the MYBPC1 phenotypic spectrum: a novel homozygous mutation causes arthrogryposis multiplex congenita. ['MYBPC1'] 9\n",
      "163\n",
      "155\n",
      "152\n",
      "158\n",
      "154\n",
      "Etiology of genetic muscle disorders induced by mutations in fast and slow skeletal MyBP-C paralogs. ['MYBPC1'] 2\n",
      "168\n",
      "148\n",
      "158\n",
      "A Case Series of Patients With MYBPC1 Gene Variants Featuring Undulating Tongue Movements as Myogenic Tremor. ['MYBPC1'] 0\n",
      "149\n",
      "Congenital tremor and myopathy secondary to novel MYBPC1 variant. ['MYBPC1'] 0\n",
      "The number of title matching paper: 8\n",
      "486\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "8. MYBPC1, myosin-binding protein C, slow-type, is a structural component of the muscle sarcomere that contributes to the regulation of muscle contraction.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Distal arthrogryposis type I (DA1) is a disorder characterized by congenital contractures of the hands and feet for which few genes have been identified. Here we describe a five-generation family with DA1 segregating as an autosomal dominant disorder with complete penetrance. Genome-wide linkage analysis using Affymetrix GeneChip Mapping 10K data from 12 affected members of this family revealed a multipoint LOD(max) of 3.27 on chromosome 12q. Sequencing of the slow-twitch skeletal muscle myosin binding protein C1 (MYBPC1), located within the linkage interval, revealed a missense mutation (c.706T>C) that segregated with disease in this family and causes a W236R amino acid substitution. A second MYBPC1 missense mutation was identified (c.2566T>C)(Y856H) in another family with DA1, accounting for an MYBPC1 mutation frequency of 13% (two of 15). Skeletal muscle biopsies from affected patients showed type I (slow-twitch) fibers were smaller than type II fibers. Expression of a green fluorescent protein (GFP)-tagged MYBPC1 construct containing WT and DA1 mutations in mouse skeletal muscle revealed robust sarcomeric localization. In contrast, a more diffuse localization was seen when non-fused GFP and MYBPC1 proteins containing corresponding MYBPC3 amino acid substitutions (R326Q, E334K) that cause hypertrophic cardiomyopathy were expressed. These findings reveal that the MYBPC1 is a novel gene responsible for DA1, though the mechanism of disease may differ from how some cardiac MYBPC3 mutations cause hypertrophic cardiomyopathy.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  MYBPC1\n",
      "Utilizing benchmarked dataset and gene regulatory network to investigate hub genes in postmenopausal osteoporosis.\n",
      "414\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "8. MYBPC1, myosin-binding protein C, slow-type, is a structural component of the muscle sarcomere that contributes to the regulation of muscle contraction.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Muscle contraction requires high energy fluxes, which are supplied by MM-CK (muscle-type creatine kinase) which couples to the myofibril. However, little is known about the detailed molecular mechanisms of how MM-CK participates in and is regulated during muscle contraction. In the present study, MM-CK is found to physically interact with the slow skeletal muscle-type MyBPC1 (myosin-binding protein C1). The interaction between MyBPC1 and MM-CK depended on the creatine concentration in a dose-dependent manner, but not on ATP, ADP or phosphocreatine. The MyBPC1-CK interaction favoured acidic conditions, and the two molecules dissociated at above pH 7.5. Domain-mapping experiments indicated that MM-CK binds to the C-terminal domains of MyBPC1, which is also the binding site of myosin. The functional coupling of myosin, MyBPC1 and MM-CK is further corroborated using an ATPase activity assay in which ATP expenditure accelerates upon the association of the three proteins, and the apparent K(m) value of myosin is therefore reduced. The results of the present study suggest that MyBPC1 acts as an adaptor to connect the ATP consumer (myosin) and the regenerator (MM-CK) for efficient energy metabolism and homoeostasis.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "348\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "8. MYBPC1, myosin-binding protein C, slow-type, is a structural component of the muscle sarcomere that contributes to the regulation of muscle contraction.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Cardiac myosin binding protein C (cMyBP-C) is an integral sarcomeric protein that associates with the thick, thin, and titin filament systems in the contractile apparatus. Three different isoforms of MyBP-C exist in mammalian muscle: slow skeletal (MyBPC1), fast skeletal (MyBP-C2, with several variants), and cardiac (cMyBP-C). Genetic screening studies show that mutations in MYBPC3 occur frequently and are responsible for as many as 30-35 % of identified cases of familial hypertrophic cardiomyopathy. The function of cMyBP-C is stringently regulated by its post-translational modification. In particular, the addition of phosphate groups occurs with high frequency on certain serine residues that are located in the cardiac-specific regulatory M domain. Phosphorylation of this domain has been extensively studied in vitro and in vivo. Phosphorylation of the M domain can regulate the manner in which actin and myosin interact, affecting the cross bridge cycle and ultimately, cardiac hemodynamics.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 8, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "9. RTN1, reticulon 1, is involved in the endoplasmic reticulum structure and may have a role in neurodegenerative diseases.\n",
      "===========================================================================\n",
      "172\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "9. RTN1, reticulon 1, is involved in the endoplasmic reticulum structure and may have a role in neurodegenerative diseases.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "RTN1\n",
      "292\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "9. RTN1, reticulon 1, is involved in the endoplasmic reticulum structure and may have a role in neurodegenerative diseases.\n",
      "    \n",
      "Result:\n",
      "endoplasmic reticulum,neurodegenerative diseases,reticulon\n",
      "PubMed Keywords:  (\"RTN1\") AND (\"endoplasmic reticulum\" OR \"neurodegenerative diseases\" OR \"reticulon\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "147\n",
      "148\n",
      "Reticulon proteins: emerging players in neurodegenerative diseases. ['RTN1'] 43\n",
      "167\n",
      "141\n",
      "Gene networks in neurodegenerative disorders. ['RTN1'] 27\n",
      "155\n",
      "RTN1 and RTN3 protein are differentially associated with senile plaques in Alzheimer's brains. ['RTN1'] 21\n",
      "151\n",
      "Reticulon1-C modulates protein disulphide isomerase function. ['RTN1'] 13\n",
      "166\n",
      "Reticulon-1A mediates diabetic kidney disease progression through endoplasmic reticulum-mitochondrial contacts in tubular epithelial cells. ['RTN1'] 11\n",
      "151\n",
      "Reticulons as markers of neurological diseases: focus on amyotrophic lateral sclerosis. ['RTN1'] 7\n",
      "152\n",
      "168\n",
      "147\n",
      "Identification of MANF as a protein interacting with RTN1-C. ['RTN1'] 5\n",
      "165\n",
      "160\n",
      "160\n",
      "162\n",
      "Reticulons bind sphingolipids to activate the endoplasmic reticulum cell cycle checkpoint, the ER surveillance pathway. ['RTN1'] 0\n",
      "The number of title matching paper: 8\n",
      "364\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "9. RTN1, reticulon 1, is involved in the endoplasmic reticulum structure and may have a role in neurodegenerative diseases.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Reticulons (RTNs) are a group of integral membrane proteins that have a uniquely conserved C-terminal domain named RHD. In mammalian genomes, transcripts are produced from four genes, rtn1 to rtn4, under the regulation of tissue or cell-type-specific expression. The presence of alternative promoters for gene expression and multiple cryptic splicing sites have resulted in large numbers of genes/proteins that are classified among the reticulon family. Although this family exists in almost all eukaryotes, only the rtn4 gene product, Nogo (RTN4), has gained relatively more in-depth attention. Despite predominant localization in the endoplasmic reticulum, Nogo on the cell surface appears to play a critical role as an inhibitory molecule for axonal growth and regeneration in humans and rodents. Recently, studies have expanded the biological functions of RTNs to other facets including modulating the enzymatic activity of beta-secretase in Alzheimer's disease. In this review, we summarize the accumulated findings concerning the structural and functional aspects of RTNs and speculate on their linkage to the pathogenesis of neurodegenerative diseases.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  RTN1\n",
      "The Regulatory Role of Reticulons in Neurodegeneration: Insights Underpinning Therapeutic Potential for Neurodegenerative Diseases.\n",
      "471\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "9. RTN1, reticulon 1, is involved in the endoplasmic reticulum structure and may have a role in neurodegenerative diseases.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Three neurodegenerative diseases [Amyotrophic Lateral Sclerosis (ALS), Parkinson's disease (PD) and Alzheimer's disease (AD)] have many characteristics like pathological mechanisms and genes. In this sense some researchers postulate that these diseases share the same alterations and that one alteration in a specific protein triggers one of these diseases. Analyses of gene expression may shed more light on how to discover pathways, pathologic mechanisms associated with the disease, biomarkers and potential therapeutic targets. In this review, we analyze four microarrays related to three neurodegenerative diseases. We will systematically examine seven genes (CHN1, MDH1, PCP4, RTN1, SLC14A1, SNAP25 and VSNL1) that are altered in the three neurodegenerative diseases. A network was built and used to identify pathways, miRNA and drugs associated with ALS, AD and PD using Cytoscape software an interaction network based on the protein interactions of these genes. The most important affected pathway is PI3K-Akt signalling. Thirteen microRNAs (miRNA-19B1, miRNA-107, miRNA-124-1, miRNA-124-2, miRNA-9-2, miRNA-29A, miRNA-9-3, miRNA-328, miRNA-19B2, miRNA-29B2, miRNA-124-3, miRNA-15A and miRNA-9-1) and four drugs (Estradiol, Acetaminophen, Resveratrol and Progesterone) for new possible treatments were identified.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "435\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "9. RTN1, reticulon 1, is involved in the endoplasmic reticulum structure and may have a role in neurodegenerative diseases.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Reticulon proteins (RTNs), consisting of RTN1 to RTN4, were previously shown to interact with BACE1 by negatively modulating its secretase activity. In RTN3-null mice, RTN1 expression was slightly elevated. To understand the in vivo role of RTN1, we generated RTN1-null mice and compared the effects of RTN1 and RTN3 on BACE1 modulation. We show that RTN1 is mostly expressed by neurons and not by glial cells under normal conditions, similar to the expression of RTN3. However, RTN1 is more localized in dendrites and is an excellent marker for dendrites of Purkinje cells, while RTN3 expression is less evident in dendrites. This differential localization also correlates with their associations with amyloid plaques in Alzheimer's brains: RTN3, but not RTN1, is abundantly enriched in dystrophic neurites. RTN3 deficiency causes elevation of BACE1 protein levels, while RTN1 deficiency shows no obvious effects on BACE1 activity due to compensation by RTN3, as RTN1 deficiency causes elevation of RTN3 expression. Hence, expression of RTN1 and RTN3 is tightly regulated in mouse brains. Together, our data show that RTN1 and RTN3 have differential effects on the formation of senile plaques in Alzheimer's brains and that RTN3 has a more prominent role in Alzheimer's pathogenesis.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 9, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "10. SMPDL3A, sphingomyelin phosphodiesterase acid-like 3A, is involved in lipid metabolism and signaling.\n",
      "===========================================================================\n",
      "166\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "10. SMPDL3A, sphingomyelin phosphodiesterase acid-like 3A, is involved in lipid metabolism and signaling.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "SMPDL3A\n",
      "276\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "10. SMPDL3A, sphingomyelin phosphodiesterase acid-like 3A, is involved in lipid metabolism and signaling.\n",
      "    \n",
      "Result:\n",
      "lipid metabolism,sphingomyelin,signaling\n",
      "PubMed Keywords:  (\"SMPDL3A\") AND (\"lipid metabolism\" OR \"sphingomyelin\" OR \"signaling\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "10 references are queried\n",
      "164\n",
      "Sphingomyelin phosphodiesterase acid-like 3A (SMPDL3A) is a novel nucleotide phosphodiesterase regulated by cholesterol in human macrophages. ['SMPDL3A'] 14\n",
      "154\n",
      "Regulation of sphingomyelin phosphodiesterase acid-like 3A gene (SMPDL3A) by liver X receptors. ['SMPDL3A'] 11\n",
      "161\n",
      "N-glycosylation of human sphingomyelin phosphodiesterase acid-like 3A (SMPDL3A) is essential for stability, secretion and activity. ['SMPDL3A'] 5\n",
      "155\n",
      "Structural Basis for Nucleotide Hydrolysis by the Acid Sphingomyelinase-like Phosphodiesterase SMPDL3A. ['SMPDL3A'] 5\n",
      "166\n",
      "The structure and catalytic mechanism of human sphingomyelin phosphodiesterase like 3a--an acid sphingomyelinase homologue with a novel nucleotide hydrolase activity. ['SMPDL3A'] 5\n",
      "166\n",
      "Ligand-dependent and -independent regulation of human hepatic sphingomyelin phosphodiesterase acid-like 3A expression by pregnane X receptor and crosstalk with liver X receptor. ['SMPDL3A'] 4\n",
      "162\n",
      "142\n",
      "SMPDL3A links cholesterol metabolism to the cGAS-STING pathway. ['SMPDL3A'] 0\n",
      "The number of title matching paper: 7\n",
      "423\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "10. SMPDL3A, sphingomyelin phosphodiesterase acid-like 3A, is involved in lipid metabolism and signaling.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Cholesterol-loaded foam cell macrophages are prominent in atherosclerotic lesions and play complex roles in both inflammatory signaling and lipid metabolism, which are underpinned by large scale reprogramming of gene expression. We performed a microarray study of primary human macrophages that showed that transcription of the sphingomyelin phosphodiesterase acid-like 3A (SMPDL3A) gene is up-regulated after cholesterol loading. SMPDL3A protein expression in and secretion from primary macrophages are stimulated by cholesterol loading, liver X receptor ligands, and cyclic AMP, and N-glycosylated SMPDL3A protein is detectable in circulating blood. We demonstrate for the first time that SMPDL3A is a functional phosphodiesterase with an acidic pH optimum. We provide evidence that SMPDL3A is not an acid sphingomyelinase but unexpectedly is active against nucleotide diphosphate and triphosphate substrates at acidic and neutral pH. SMPDL3A is a major source of nucleotide phosphodiesterase activity secreted by liver X receptor-stimulated human macrophages. Extracellular nucleotides such as ATP may activate pro-inflammatory responses in immune cells. Increased expression and secretion of SMPDL3A by cholesterol-loaded macrophage foam cells in lesions may decrease local concentrations of pro-inflammatory nucleotides and potentially represent a novel anti-inflammatory axis linking lipid metabolism with purinergic signaling in atherosclerosis.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  SMPDL3A\n",
      "A patient with compound heterozygosity of SMPD4: Another example of utility of exome-based copy number analysis in autosomal recessive disorders.\n",
      "473\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "10. SMPDL3A, sphingomyelin phosphodiesterase acid-like 3A, is involved in lipid metabolism and signaling.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Liver X receptor (LXR) α and LXRβ function as physiological sensors of cholesterol metabolites (oxysterols), regulating key genes involved in cholesterol and lipid metabolism. LXRs have been extensively studied in both human and rodent cell systems, revealing their potential therapeutic value in the contexts of atherosclerosis and inflammatory diseases. The LXR genome landscape has been investigated in murine macrophages but not in human THP-1 cells, which represent one of the frequently used monocyte/macrophage cell systems to study immune responses. We used a whole-genome screen to detect direct LXR target genes in THP-1 cells treated with two widely used LXR ligands [N-(2,2,2-trifluoroethyl)-N-[4-[2,2,2-trifluoro-1-hydroxy-1-(trifluoromethyl)-ethyl]phenyl]-benzenesulfonamide (T0901317) and 3-[3-[N-(2-chloro-3-trifluoromethylbenzyl)-(2,2-diphenylethyl)amino]propyloxy] phenylacetic acid hydrochloride (GW3965)]. This screen identified the sphingomyelin phosphodiesterase acid-like 3A (SMPDL3A) gene as a novel LXR-regulated gene, with an LXR response element within its promoter. We investigated the regulation of SMPDL3A gene expression by LXRs across several human and mouse cell types. These studies indicate that the induction of SMPDL3A is LXR-dependent and is restricted to human blood cells with no induction observed in mouse cellular systems.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "592\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "10. SMPDL3A, sphingomyelin phosphodiesterase acid-like 3A, is involved in lipid metabolism and signaling.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Sphingomyelin phosphodiesterase acid-like 3A (SMPDL3A) is a recently identified phosphodiesterase, which is a secreted <i>N</i>-linked glycoprotein. SMPDL3A is highly homologous to acid sphingomyelinase (aSMase), but unlike aSMase cannot cleave sphingomyelin. Rather, SMPDL3A hydrolyzes nucleotide tri- and diphosphates and their derivatives. While recent structural studies have shed light on these unexpected substrate preferences, many other aspects of SMPDL3A biology, which may give insight into its function <i>in vivo</i>, remain obscure. Here, we investigate the roles of N-glycosylation in the expression, secretion and activity of human SMPDL3A, using inhibitors of N-glycosylation and site-directed mutagenesis, with either THP-1 macrophages or CHO cells expressing human SMPDL3A. Tunicamycin (TM) treatment resulted in expression of non-glycosylated SMPDL3A that was not secreted, and was largely degraded by the proteasome. Proteasomal inhibition restored levels of SMPDL3A in TM-treated cells, although this non-glycosylated protein lacked phosphodiesterase activity. Enzymatic deglycosylation of purified recombinant SMPDL3A also resulted in significant loss of phosphodiesterase activity. Site-directed mutagenesis of individual N-glycosylation sites in SMPDL3A identified glycosylation of Asn69 and Asn222 as affecting maturation of its <i>N</i>-glycans and secretion. Glycosylation of Asn356 in SMPDL3A, an <i>N-</i>linked site conserved throughout the aSMase-like family, was critical for protection against proteasomal degradation and preservation of enzymatic activity. We provide the first experimental evidence for a predicted 22 residue N-terminal signal peptide in SMPDL3A, which is essential for facilitating glycosylation and is removed from the mature protein secreted from CHO cells. In conclusion, site-specific N-glycosylation is essential for the intracellular stability, secretion and activity of human SMPDL3A.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 10, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "11. CD24 is a cell adhesion molecule that has a role in the regulation of cell proliferation and differentiation, and is implicated in cancer progression.\n",
      "===========================================================================\n",
      "164\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "11. CD24 is a cell adhesion molecule that has a role in the regulation of cell proliferation and differentiation, and is implicated in cancer progression.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "CD24\n",
      "275\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "11. CD24 is a cell adhesion molecule that has a role in the regulation of cell proliferation and differentiation, and is implicated in cancer progression.\n",
      "    \n",
      "Result:\n",
      "cell adhesion,cancer progression,cell proliferation\n",
      "PubMed Keywords:  (\"CD24\") AND (\"cell adhesion\" OR \"cancer progression\" OR \"cell proliferation\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "143\n",
      "143\n",
      "138\n",
      "174\n",
      "136\n",
      "142\n",
      "148\n",
      "Involvement of CD24 in Multiple Cancer Related Pathways Makes It an Interesting New Target for Cancer Therapy. ['CD24'] 23\n",
      "157\n",
      "140\n",
      "144\n",
      "CD24, A Review of its Role in Tumor Diagnosis, Progression and Therapy. ['CD24'] 15\n",
      "141\n",
      "140\n",
      "CD24 is a genetic modifier for risk and progression of prostate cancer. ['CD24'] 6\n",
      "143\n",
      "153\n",
      "139\n",
      "Targeting CD24 as a novel immunotherapy for solid cancers. ['CD24'] 0\n",
      "The number of title matching paper: 4\n",
      "365\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "11. CD24 is a cell adhesion molecule that has a role in the regulation of cell proliferation and differentiation, and is implicated in cancer progression.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "CD24 (cluster of differentiation 24) is a small heavy glycosylated protein, which is overexpressed in many cancer and some cancer stem cells and is associated with the development, invasion, and metastasis of cancer cells. The exact role of CD24 in these processes is not fully understood, however, in this article, it has been tried to present a collection of cancer-related mechanisms attributed to CD24. Based on the literature, CD24 dis-regulates different signaling pathways in various cancer cells, including; Src kinases, STAT3, EGFR, Wnt/β-catenin and MAPK. Src kinases play an important role in the signaling pathways which activate p38 MAPK and STAT3 pathways. Akt and ERK are downstream effectors of CD24-activated EGFR, which promote cell proliferation, invasion and metastasis. CD24 increases the expression of HER2 by the activation of NF-κB transcription factor. Moreover, CD24 up-regulates the expression of miR-21 oncomir through the activation of Src kinases. Identification of the details of these pathways and also new pathways will help researchers to explore new CD24 targeted therapies.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  CD24\n",
      "13-Oxo-9Z,11E-octadecadienoic Acid Down-regulates c-Myc and Attenuates Breast Cancer Cell Stemness.\n",
      "307\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "11. CD24 is a cell adhesion molecule that has a role in the regulation of cell proliferation and differentiation, and is implicated in cancer progression.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "CD24, is a mucin-like GPI-anchored molecules. By immunohistochemistry, it is widely detected in many solid tumors, such as breast cancers, genital system cancers, digestive system cancers, neural system cancers and so on. The functional roles of CD24 are either fulfilled by combination with ligands or participate in signal transduction, which mediate the initiation and progression of neoplasms. However, the character of CD24 remains to be intriguing because there are still opposite voices about the impact of CD24 on tumors. In preclinical studies, CD24 target therapies, including monoclonal antibodies, target silencing by RNA interference and immunotherapy, have shown us brighten futures on the anti-tumor application. Nevertheless, evidences based on clinical studies are urgently needed. Here, with expectancy to spark new ideas, we summarize the relevant studies about CD24 from a tumor perspective.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "594\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "11. CD24 is a cell adhesion molecule that has a role in the regulation of cell proliferation and differentiation, and is implicated in cancer progression.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "CD24 plays an oncogenic role in the onset and progression of various human cancers, including prostate cancer. In the present study, we identified two linkage disequilibrium blocks with four recombination hotspot motifs in human CD24 locus. To elucidate whether genetic variants of CD24 are associated with susceptibility to prostate cancer and its disease status, we conducted a case-control association study with two P170 C/T and P-534 A/C polymorphisms of CD24 in 590 patients with prostate cancer and 590 healthy controls. A significant increased risk of prostate cancer was found in men with the P170<sup>T/T</sup> genotype over the P170<sup>C/C</sup> genotype (odd ratio = 1.74, 95% confidence interval = 1.16-2.63, P = 0.008), and in men with the P-534<sup>C/C</sup> genotype over the P-534<sup>A/A</sup> genotype (odd ratio = 1.47, 95% CI = 1.18-2.26, P = 0.003). Cochran-Armitage trend analysis showed that the P170<sup>T</sup> allele was significantly correlated with an increased risk of prostate cancer progression (P = 0.029, trend between genotypes and stages) and this observation was also validated in an independent sample cohort. Next, we found that tumors with P170<sup>T</sup> or P-534<sup>C</sup> alleles had more twofold increased protein expressions of CD24 as compared to those with P170<sup>C</sup> or P-534<sup>A</sup> alleles, respectively. Likewise, tumors with a combination of P170<sup>T/T</sup> and P-534<sup>C/C</sup> genotypes were associated with a high mRNA level of CD24. Our data suggest a significant association of CD24 genetic variants with prostate cancer onset and progression, which provides new insight into molecular genetics of prostate cancer; however, these findings need to be validated in multiple independent cohorts. © 2016 Wiley Periodicals, Inc.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 11, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "12. EPCAM, epithelial cell adhesion molecule, is involved in cell signaling, cell adhesion, proliferation, and differentiation, and is often used as a marker for cancer stem cells.\n",
      "===========================================================================\n",
      "175\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "12. EPCAM, epithelial cell adhesion molecule, is involved in cell signaling, cell adhesion, proliferation, and differentiation, and is often used as a marker for cancer stem cells.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "EPCAM\n",
      "286\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "12. EPCAM, epithelial cell adhesion molecule, is involved in cell signaling, cell adhesion, proliferation, and differentiation, and is often used as a marker for cancer stem cells.\n",
      "    \n",
      "Result:\n",
      "cell adhesion,cell signaling,cancer stem cells\n",
      "PubMed Keywords:  (\"EPCAM\") AND (\"cell adhesion\" OR \"cell signaling\" OR \"cancer stem cells\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "157\n",
      "Expression and function of epithelial cell adhesion molecule EpCAM: where are we after 40 years? ['EPCAM'] 95\n",
      "150\n",
      "EpCAM and the biology of hepatic stem/progenitor cells. ['EPCAM'] 60\n",
      "151\n",
      "Cancer stem cell-targeted bio-imaging and chemotherapeutic perspective. ['EPCAM'] 46\n",
      "160\n",
      "EpCAM-high liver cancer stem cells resist natural killer cell-mediated cytotoxicity by upregulating CEACAM1. ['EPCAM'] 45\n",
      "152\n",
      "Antibody Based EpCAM Targeted Therapy of Cancer, Review and Update. ['EPCAM'] 44\n",
      "153\n",
      "The Identifications and Clinical Implications of Cancer Stem Cells in Colorectal Cancer. ['EPCAM'] 42\n",
      "147\n",
      "Cancer stem cells in human digestive tract malignancies. ['EPCAM'] 31\n",
      "148\n",
      "Shedding light on the EpCAM: An overview. ['EPCAM'] 21\n",
      "145\n",
      "Heterogeneity of Hepatic Cancer Stem Cells. ['EPCAM'] 17\n",
      "142\n",
      "151\n",
      "EpCAM associates with integrin and regulates cell adhesion in cancer cells. ['EPCAM'] 11\n",
      "144\n",
      "Current View on EpCAM Structural Biology. ['EPCAM'] 10\n",
      "156\n",
      "BMP9-ID1 signaling promotes EpCAM-positive cancer stem cell properties in hepatocellular carcinoma. ['EPCAM'] 8\n",
      "154\n",
      "Cellular and Molecular Biology of Cancer Stem Cells of Hepatocellular Carcinoma. ['EPCAM'] 7\n",
      "148\n",
      "147\n",
      "EpCAM as Modulator of Tissue Plasticity. ['EPCAM'] 4\n",
      "153\n",
      "Enrichment of Cancer Cells Based on Antibody-Free Selective Cell Adhesion. ['EPCAM'] 1\n",
      "The number of title matching paper: 15\n",
      "451\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "12. EPCAM, epithelial cell adhesion molecule, is involved in cell signaling, cell adhesion, proliferation, and differentiation, and is often used as a marker for cancer stem cells.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "EpCAM (epithelial cell adhesion molecule) was discovered four decades ago as a tumor antigen on colorectal carcinomas. Owing to its frequent and high expression on carcinomas and their metastases, EpCAM serves as a prognostic marker, a therapeutic target, and an anchor molecule on circulating and disseminated tumor cells (CTCs/DTCs), which are considered the major source for metastatic cancer cells. Today, EpCAM is reckoned as a multi-functional transmembrane protein involved in the regulation of cell adhesion, proliferation, migration, stemness, and epithelial-to-mesenchymal transition (EMT) of carcinoma cells. To fulfill these functions, EpCAM is instrumental in intra- and intercellular signaling as a full-length molecule and following regulated intramembrane proteolysis, generating functionally active extra- and intracellular fragments. Intact EpCAM and its proteolytic fragments interact with claudins, CD44, E-cadherin, epidermal growth factor receptor (EGFR), and intracellular signaling components of the WNT and Ras/Raf pathways, respectively. This plethora of functions contributes to shaping intratumor heterogeneity and partial EMT, which are major determinants of the clinical outcome of carcinoma patients. EpCAM represents a marker for the epithelial status of primary and systemic tumor cells and emerges as a measure for the metastatic capacity of CTCs. Consequentially, EpCAM has reclaimed potential as a prognostic marker and target on primary and systemic tumor cells.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  EPCAM\n",
      "Proteomics of cancer stem cells.\n",
      "472\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "12. EPCAM, epithelial cell adhesion molecule, is involved in cell signaling, cell adhesion, proliferation, and differentiation, and is often used as a marker for cancer stem cells.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Epithelial cell adhesion molecule (EpCAM) is a transmembrane glycoprotein, which is frequently and highly expressed on carcinomas, tumor-initiating cells, selected tissue progenitors, and embryonic and adult stem cells. During liver development, EpCAM demonstrates a dynamic expression, since it can be detected in fetal liver, including cells of the parenchyma, whereas mature hepatocytes are devoid of EpCAM. Liver regeneration is associated with a population of EpCAM-positive cells within ductular reactions, which gradually lose the expression of EpCAM along with maturation into hepatocytes. EpCAM can be switched on and off through a wide panel of strategies to fine-tune EpCAM-dependent functional and differentiative traits. EpCAM-associated functions relate to cell-cell adhesion, proliferation, maintenance of a pluripotent state, regulation of differentiation, migration, and invasion. These functions can be conferred by the full-length protein and/or EpCAM-derived fragments, which are generated upon regulated intramembrane proteolysis. Control by EpCAM therefore not only depends on the presence of full-length EpCAM at cellular membranes but also on varying rates of the formation of EpCAM-derived fragments that have their own regulatory properties and on changes in the association of EpCAM with interaction partners. Thus spatiotemporal localization of EpCAM in immature liver progenitors, transit-amplifying cells, and mature liver cells will decisively impact the regulation of EpCAM functions and might be one of the triggers that contributes to the adaptive processes in stem/progenitor cell lineages. This review will summarize EpCAM-related molecular events and how they relate to hepatobiliary differentiation and regeneration.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "406\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "12. EPCAM, epithelial cell adhesion molecule, is involved in cell signaling, cell adhesion, proliferation, and differentiation, and is often used as a marker for cancer stem cells.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Cancer stem cells (CSCs), also called tumor-initiating cells (TICs), have been studied intensively due to their rapid proliferation, migration, and role in the recurrence of cancer. In general, CSC marker-positive cells [CD133, CD44, CD166, aldehyde dehydrogenase (ALDH), and epithelial cell adhesion molecule (EpCAM)] exhibit a 100-fold increased capacity to initiate cancer. Within a heterogeneous tumor mass, only approximately 0.05-3% of cells are suspected to be CSCs and able to proliferate under hypoxia. Interestingly, CSCs, cancer cells, and normal stem cells share many cytochemical properties, such as inhibition of the redox system for reactive oxygen species (ROS) production and high expression of drug resistance transporters. However, compared to normal stem cells, CSCs develop unique metabolic flexibility, which involves switching between oxidative phosphorylation (OXPHOS) and glycolysis as their main source of energy. Due to the similarities between CSCs and other cancer cells and normal stem cells, limited chemotherapeutic and bio-imaging reagents specific for CSCs have been developed. In this short review, we address the current knowledge regarding CSCs with a focus on designing chemotherapeutic and bio-imaging reagents that target CSCs.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 12, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "13. LMO2, LIM domain only 2, is a transcription factor implicated in the regulation of hematopoiesis and angiogenesis, and is associated with certain types of leukemia.\n",
      "===========================================================================\n",
      "174\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "13. LMO2, LIM domain only 2, is a transcription factor implicated in the regulation of hematopoiesis and angiogenesis, and is associated with certain types of leukemia.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "LMO2\n",
      "287\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "13. LMO2, LIM domain only 2, is a transcription factor implicated in the regulation of hematopoiesis and angiogenesis, and is associated with certain types of leukemia.\n",
      "    \n",
      "Result:\n",
      "hematopoiesis,transcription factor,angiogenesis\n",
      "PubMed Keywords:  (\"LMO2\") AND (\"hematopoiesis\" OR \"transcription factor\" OR \"angiogenesis\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "152\n",
      "LMO2 at 25 years: a paradigm of chromosomal translocation proteins. ['LMO2'] 34\n",
      "152\n",
      "158\n",
      "The LIM-only transcription factor LMO2 determines tumorigenic and angiogenic traits in glioma stem cells. ['LMO2'] 33\n",
      "161\n",
      "GATA2 and Lmo2 control angiogenesis and lymphangiogenesis via direct transcriptional regulation of neuropilin-2. ['LMO2'] 27\n",
      "157\n",
      "164\n",
      "172\n",
      "Lmo2 (LIM-Domain-Only 2) Modulates Sphk1 (Sphingosine Kinase) and Promotes Endothelial Cell Migration. ['LMO2'] 17\n",
      "159\n",
      "LMO2 activation by deacetylation is indispensable for hematopoiesis and T-ALL leukemogenesis. ['LMO2'] 14\n",
      "170\n",
      "156\n",
      "Roles of the RUNX1 Enhancer in Normal Hematopoiesis and Leukemogenesis. ['LMO2'] 7\n",
      "162\n",
      "Fli1<sup>+</sup> cells transcriptional analysis reveals an Lmo2-Prdm16 axis in angiogenesis. ['LMO2'] 7\n",
      "170\n",
      "156\n",
      "Genome-wide transcription factor-binding maps reveal cell-specific changes in the regulatory architecture of human HSPCs. ['LMO2'] 2\n",
      "161\n",
      "146\n",
      "146\n",
      "154\n",
      "The number of title matching paper: 8\n",
      "431\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "13. LMO2, LIM domain only 2, is a transcription factor implicated in the regulation of hematopoiesis and angiogenesis, and is associated with certain types of leukemia.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "LMO2 was first discovered through proximity to frequently occurring chromosomal translocations in T cell acute lymphoblastic leukaemia (T-ALL). Subsequent studies on its role in tumours and in normal settings have highlighted LMO2 as an archetypical chromosomal translocation oncogene, activated by association with antigen receptor gene loci and a paradigm for translocation gene activation in T-ALL. The normal function of LMO2 in haematopoietic cell fate and angiogenesis suggests it is a master gene regulator exerting a dysfunctional control on differentiation following chromosomal translocations. Its importance in T cell neoplasia has been further emphasized by the recurrent findings of interstitial deletions of chromosome 11 near LMO2 and of LMO2 as a target of retroviral insertion gene activation during gene therapy trials for X chromosome-linked severe combined immuno-deficiency syndrome, both types of event leading to similar T cell leukaemia. The discovery of LMO2 in some B cell neoplasias and in some epithelial cancers suggests a more ubiquitous function as an oncogenic protein, and that the current development of novel inhibitors will be of great value in future cancer treatment. Further, the role of LMO2 in angiogenesis and in haematopoietic stem cells (HSCs) bodes well for targeting LMO2 in angiogenic disorders and in generating autologous induced HSCs for application in various clinical indications.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  LMO2\n",
      "Stem cell transcription factors.\n",
      "516\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "13. LMO2, LIM domain only 2, is a transcription factor implicated in the regulation of hematopoiesis and angiogenesis, and is associated with certain types of leukemia.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Glioblastomas (GBMs) maintain their cellular heterogeneity with glioma stem cells (GSCs) producing a variety of tumor cell types. Here we interrogated the oncogenic roles of Lim domain only 2 (LMO2) in GBM and GSCs in mice and human. High expression of LMO2 was found in human patient-derived GSCs compared with the differentiated progeny cells. LMO2 is required for GSC proliferation both in vitro and in vivo, as shRNA-mediated LMO2 silencing attenuated tumor growth derived from human GSCs. Further, LMO2 is sufficient to induce stem cell characteristics (stemness) in mouse premalignant astrocytes, as forced LMO2 expression facilitated in vitro and in vivo growth of astrocytes derived from Ink4a/Arf null mice and acquisition of GSC phenotypes. A subset of mouse and human GSCs converted into vascular endothelial-like tumor cells both in vitro and in vivo, which phenotype was attenuated by LMO2 silencing and promoted by LMO2 overexpression. Mechanistically, the action of LMO2 for induction of glioma stemness is mediated by transcriptional regulation of Jagged1 resulting in activation of the Notch pathway, whereas LMO2 directly occupies the promoter regions of the VE-cadherin gene for a gain of endothelial cellular phenotype. Subsequently, selective ablation of human GSC-derived VE-cadherin-expressing cells attenuated vascular formation in mouse intracranial tumors, thereby significantly prolonging mouse survival. Clinically, LMO2 expression was elevated in GBM tissues and inversely correlated with prognosis of GBM patients. Taken together, our findings describe novel dual roles of LMO2 to induce tumorigenesis and angiogenesis, and provide potential therapeutic targets in GBMs.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "473\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "13. LMO2, LIM domain only 2, is a transcription factor implicated in the regulation of hematopoiesis and angiogenesis, and is associated with certain types of leukemia.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "GATA-binding protein 2 (GATA2) and LIM domain only 2 (Lmo2) form common transcription complexes during hematopoietic differentiation. Here we show that these two transcription factors also play a key role in endothelial cells (EC) and lymphatic EC (LEC) function. Primary EC and tumor-associated blood vessels expressed GATA2 and Lmo2. VEGF-induced sprouting angiogenesis in both differentiating embryonic stem cells (embryoid bodies) and primary EC increased GATA2 and Lmo2 levels. Conversely, silencing of GATA2 and Lmo2 expression in primary EC inhibited VEGF-induced angiogenic activity, including EC migration and sprouting in vitro, two key steps of angiogenesis in vivo. This inhibition of EC function was associated with downregulated expression of neuropilin-2 (NRP2), a co-receptor of VEGFRs for VEGF, at the protein, mRNA and promoter levels. NRP2 overexpression partially rescued the impaired angiogenic sprouting in the GATA2/Lmo2 knockdown EC, confirming that GATA2 and Lmo2 mediated EC function, at least in part, by directly regulating NRP2 gene expression. Furthermore, it was found that primary LEC expressed GATA2 and Lmo2 as well. Silencing of GATA2 and Lmo2 expression in LEC inhibited VEGF-induced LEC sprouting, also in a NRP2-dependent manner. In conclusion, our results demonstrate that GATA2 and Lmo2 cooperatively regulate VEGF-induced angiogenesis and lymphangiogenesis via NRP2.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 13, 3 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "14. NREP, neuronal regeneration related protein, is involved in cellular growth and differentiation, particularly in the nervous system.\n",
      "===========================================================================\n",
      "158\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "14. NREP, neuronal regeneration related protein, is involved in cellular growth and differentiation, particularly in the nervous system.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "NREP\n",
      "272\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "14. NREP, neuronal regeneration related protein, is involved in cellular growth and differentiation, particularly in the nervous system.\n",
      "    \n",
      "Result:\n",
      "neuronal regeneration,cellular differentiation,nervous system\n",
      "PubMed Keywords:  (\"NREP\") AND (\"neuronal regeneration\" OR \"cellular differentiation\" OR \"nervous system\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "5 references are queried\n",
      "146\n",
      "150\n",
      "141\n",
      "The number of title matching paper: 0\n",
      "In paragraph 14, 0 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "15. WFDC2, also known as HE4, is a protein with a putative role in natural immunity and is used as a biomarker for ovarian cancer.\n",
      "===========================================================================\n",
      "169\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "15. WFDC2, also known as HE4, is a protein with a putative role in natural immunity and is used as a biomarker for ovarian cancer.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "WFDC2\n",
      "280\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "15. WFDC2, also known as HE4, is a protein with a putative role in natural immunity and is used as a biomarker for ovarian cancer.\n",
      "    \n",
      "Result:\n",
      "natural immunity,ovarian cancer,biomarker\n",
      "PubMed Keywords:  (\"WFDC2\") AND (\"natural immunity\" OR \"ovarian cancer\" OR \"biomarker\") AND (hasabstract[text]) AND humans[mh] NOT Retracted Publication[pt]\n",
      "Serching paper with keywords...\n",
      "20 references are queried\n",
      "140\n",
      "148\n",
      "142\n",
      "148\n",
      "164\n",
      "A bioinformatic analysis of WFDC2 (HE4) expression in high grade serous ovarian cancer reveals tumor-specific changes in metabolic and extracellular matrix gene expression. ['WFDC2'] 1\n",
      "154\n",
      "147\n",
      "[Expression and significance of WFDC2 in ovarian cancer based on multiple gene databases]. ['WFDC2'] 0\n",
      "The number of title matching paper: 2\n",
      "492\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "15. WFDC2, also known as HE4, is a protein with a putative role in natural immunity and is used as a biomarker for ovarian cancer.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Human epididymis protein-4 (HE4/WFDC2) has been well-studied as an ovarian cancer clinical biomarker. To improve our understanding of its functional role in high grade serous ovarian cancer, we determined transcriptomic differences between ovarian tumors with high- versus low-WFDC2 mRNA levels in The Cancer Genome Atlas dataset. High-WFDC2 transcript levels were significantly associated with reduced survival in stage III/IV serous ovarian cancer patients. Differential expression and correlation analyses revealed secretory leukocyte peptidase inhibitor (SLPI/WFDC4) as the gene most positively correlated with WFDC2, while A kinase anchor protein-12 was most negatively correlated. WFDC2 and SLPI were strongly correlated across many cancers. Gene ontology analysis revealed enrichment of oxidative phosphorylation in differentially expressed genes associated with high-WFDC2 levels, while extracellular matrix organization was enriched among genes associated with low-WFDC2 levels. Immune cell subsets found to be positively correlated with WFDC2 levels were B cells and plasmacytoid dendritic cells, while neutrophils and endothelial cells were negatively correlated with WFDC2. Results were compared with DepMap cell culture gene expression data. Gene ontology analysis of k-means clustering revealed that genes associated with low-WFDC2 were also enriched in extracellular matrix and adhesion categories, while high-WFDC2 genes were enriched in epithelial cell proliferation and peptidase activity. These results support previous findings regarding the effect of HE4/WFDC2 on ovarian cancer pathogenesis in cell lines and mouse models, while adding another layer of complexity to its potential functions in ovarian tumor tissue. Further experimental explorations of these findings in the context of the tumor microenvironment are merited.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  WFDC2\n",
      "HE4-test of urine and body fluids for diagnosis of gynecologic cancer.\n",
      "416\n",
      "Query:  \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "15. WFDC2, also known as HE4, is a protein with a putative role in natural immunity and is used as a biomarker for ovarian cancer.\n",
      "\n",
      "and an abstract.\n",
      "Abstract:\n",
      "Objective To investigate the expression and clinical significance of whey acidic protein (WAP) 4-disulfide core domain 2 (WFDC2) in ovarian cancer. Methods The expression of WFDC2 gene in normal human tissues, ovarian cancer tissues and ovarian cancer cell lines and the correlation with the prognosis survival in patients with ovarian cancer were analyzed using the BioGPS database, Oncomine database, Cancer Cell Encyclopedia (CCLE) and Kaplan-Meier Plotter databases. Results BioGPS database analysis showed that the expression of WFDC2 was not found in normal ovarian tissues. Four hundred and seventeen samples were searched from Oncomine database, and the different expression of WFDC2 in 34 studies were statistically significant, which included the WFDC2 expression increasing in 19 studies, and WFDC2 expression decreasing in 15 studies. Meta-analysis of 7 studies meeting the setting conditions showed that the expression level of WFDC2 was high in ovarian cancer tissues. CCLE analysis showed that the expression level of WFDC2 was also high in ovarian cancer cell lines; Kaplan-Meier Plotter database results showed the overall survival time of patients with ovarian cancer in the high WFDC2 expression group was significantly longer than that in the low WFDC2 expression group. Conclusion The expression of WFDC2 in ovarian cancer tissues is high, and which is significantly associated with the prognosis survival in patients with ovarian cancer.\n",
      "\n",
      "Does this abstract support paragraph? Please tell me yes or no\n",
      "        \n",
      "Result:  Yes.\n",
      "Remained genes:  \n",
      "In paragraph 15, 2 references are matched\n",
      "\n",
      "\n",
      "Extracting keywords from paragraph\n",
      "Paragraph:\n",
      "In summary, the proteins listed are involved in a wide array of biological processes, including but not limited to enzyme regulation, cell adhesion, immune response, muscle contraction, lipid metabolism, and cellular stress response. There is no clear, singular biological process that encompasses the majority of these proteins, and they appear to represent a diverse set of functions rather than a coherent system. Therefore, the most appropriate designation for this group of proteins is a \"System of unrelated proteins\" with a confidence score of 0.00, indicating no confidence in a unified biological process.\n",
      "===========================================================================\n",
      "245\n",
      "Query:\n",
      " \n",
      "I have a paragraph\n",
      "Paragraph:\n",
      "In summary, the proteins listed are involved in a wide array of biological processes, including but not limited to enzyme regulation, cell adhesion, immune response, muscle contraction, lipid metabolism, and cellular stress response. There is no clear, singular biological process that encompasses the majority of these proteins, and they appear to represent a diverse set of functions rather than a coherent system. Therefore, the most appropriate designation for this group of proteins is a \"System of unrelated proteins\" with a confidence score of 0.00, indicating no confidence in a unified biological process.\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in this paragraph. Give me a list of gene symbols from the paragraph. Please only include genes. Return the genes as a comma separated list without spacing, if there are no genes in the statements, please return \"Unknown\" \n",
      "Result:\n",
      "Unknown\n",
      "358\n",
      "Query:\n",
      "\n",
      "I would like to search PubMed to find supporting evidence for the statements in a paragraph. Give me a maximum of 3 keywords related to the protein functions or biological processes in the statements. \n",
      "\n",
      "Example paragraph:  Involvement of pattern recognition receptors: TLR1, TLR2, and TLR3 are part of the Toll-like receptor family, which recognize pathogen-associated molecular patterns and initiate innate immune responses. NOD2 and NLRP3 are intracellular sensors that also contribute to immune activation.\n",
      "Example response: immune response,receptors,pathogen\n",
      "\n",
      "Please don't include gene symbols. Please order keywords by their importance in the paragraph, from high importance to low importance. Return the keywords as a comma separated list without spaces. If there are no keywords matching the criteria, return \"Unknown\" \n",
      "\n",
      "Please find keywords for this paragraph:\n",
      "In summary, the proteins listed are involved in a wide array of biological processes, including but not limited to enzyme regulation, cell adhesion, immune response, muscle contraction, lipid metabolism, and cellular stress response. There is no clear, singular biological process that encompasses the majority of these proteins, and they appear to represent a diverse set of functions rather than a coherent system. Therefore, the most appropriate designation for this group of proteins is a \"System of unrelated proteins\" with a confidence score of 0.00, indicating no confidence in a unified biological process.\n",
      "    \n",
      "Result:\n",
      "biological processes,enzyme regulation,cell adhesion\n",
      "No keyword generated skip referencing\n",
      "In paragraph 16, 0 references are matched\n",
      "\n",
      "\n",
      "Total 33 references are queried\n",
      "[[], ['Sørensen, K D, Abildgaard, M O, Haldrup, C, Ulhøi, B P, Kristensen, H, Strand, S, Parker, C, Høyer, S, Borre, M, Ørntoft, T F. \"Prognostic significance of aberrantly silenced ANPEP expression in prostate cancer.\" British journal of cancer, 2013, pp. 420-8, doi: https://doi.org/10.1038/bjc.2012.549', 'Shui, Li-Jun, Meng, Yan, Huang, Cun, Qian, Yi, Liu, Jia-Yin. \"Aminopeptidase N expression in the endometrium could affect endometrial receptivity.\" Biochemical and biophysical research communications, 2019, pp. 469-474, doi: https://doi.org/10.1016/j.bbrc.2019.04.174', 'Kim, Jong-Heon, Afridi, Ruqayya, Cho, Eunji, Yoon, Jong Hyuk, Lim, Yong-Hyun, Lee, Ho-Won, Ryu, Hoon, Suk, Kyoungho. \"Soluble ANPEP Released From Human Astrocytes as a Positive Regulator of Microglial Activation and Neuroinflammation: Brain Renin-Angiotensin System in Astrocyte-Microglia Crosstalk.\" Molecular & cellular proteomics : MCP, 2022, pp. 100424, doi: https://doi.org/10.1016/j.mcpro.2022.100424'], ['Shao, Shiying, Xu, QinQin, Yu, Xuefeng, Pan, Ruping, Chen, Yong. \"Dipeptidyl peptidase 4 inhibitors and their potential immune modulatory functions.\" Pharmacology & therapeutics, 2020, pp. 107503, doi: https://doi.org/10.1016/j.pharmthera.2020.107503', 'Yazbeck, Roger, Jaenisch, Simone E, Abbott, Catherine A. \"Potential disease biomarkers: dipeptidyl peptidase 4 and fibroblast activation protein.\" Protoplasma, 2018, pp. 375-386, doi: https://doi.org/10.1007/s00709-017-1129-5', 'Love, Kaitlin M, Liu, Zhenqi. \"DPP4 Activity, Hyperinsulinemia, and Atherosclerosis.\" The Journal of clinical endocrinology and metabolism, 2021, pp. 1553-1565, doi: https://doi.org/10.1210/clinem/dgab078'], [], ['Wright, C, Shin, J H, Rajpurohit, A, Deep-Soboslay, A, Collado-Torres, L, Brandon, N J, Hyde, T M, Kleinman, J E, Jaffe, A E, Cross, A J, Weinberger, D R. \"Altered expression of histamine signaling genes in autism spectrum disorder.\" Translational psychiatry, 2017, pp. e1126, doi: https://doi.org/10.1038/tp.2017.87', 'Preuss, C V, Wood, T C, Szumlanski, C L, Raftogianis, R B, Otterness, D M, Girard, B, Scott, M C, Weinshilboum, R M. \"Human histamine N-methyltransferase pharmacogenetics: common genetic polymorphisms that alter activity.\" Molecular pharmacology, 1998, pp. 708-17, doi: https://doi.org/10.1124/mol.53.4.708', 'Ogasawara, Masahito, Yamauchi, Kohei, Satoh, Yoh-Ichi, Yamaji, Ryoichi, Inui, Kenichi, Jonker, Johan W, Schinkel, Alfred H, Maeyama, Kazutaka. \"Recent advances in molecular pharmacology of the histamine systems: organic cation transporters as a histamine transporter and histamine metabolism.\" Journal of pharmacological sciences, 2006, pp. 24-30, doi: https://doi.org/10.1254/jphs.fmj06001x6'], ['Bell, Jonathan B, Eckerdt, Frank D, Alley, Kristen, Magnusson, Lisa P, Hussain, Hridi, Bi, Yingtao, Arslan, Ahmet Dirim, Clymer, Jessica, Alvarez, Angel A, Goldman, Stewart, Cheng, Shi-Yuan, Nakano, Ichiro, Horbinski, Craig, Davuluri, Ramana V, James, C David, Platanias, Leonidas C. \"MNK Inhibition Disrupts Mesenchymal Glioma Stem Cells and Prolongs Survival in a Mouse Model of Glioblastoma.\" Molecular cancer research : MCR, 2016, pp. 984-993, doi: https://doi.org/10.1158/1541-7786.MCR-16-0172'], ['Ma, Jonathan, Rubin, Bruce K, Voynow, Judith A. \"Mucins, Mucus, and Goblet Cells.\" Chest, 2018, pp. 169-176, doi: https://doi.org/10.1016/j.chest.2017.11.008', 'Okuda, Kenichi, Chen, Gang, Subramani, Durai B, Wolf, Monroe, Gilmore, Rodney C, Kato, Takafumi, Radicioni, Giorgia, Kesimer, Mehmet, Chua, Michael, Dang, Hong, Livraghi-Butrico, Alessandra, Ehre, Camille, Doerschuk, Claire M, Randell, Scott H, Matsui, Hirotoshi, Nagase, Takahide, O\\'Neal, Wanda K, Boucher, Richard C. \"Localization of Secretory Mucins MUC5AC and MUC5B in Normal/Healthy Human Airways.\" American journal of respiratory and critical care medicine, 2019, pp. 715-727, doi: https://doi.org/10.1164/rccm.201804-0734OC', 'Ridley, Caroline, Thornton, David J. \"Mucins: the frontline defence of the lung.\" Biochemical Society transactions, 2018, pp. 1099-1106, doi: https://doi.org/10.1042/BST20170402'], ['Gurnett, Christina A, Desruisseau, David M, McCall, Kevin, Choi, Ryan, Meyer, Zachary I, Talerico, Michael, Miller, Sara E, Ju, Jeong-Sun, Pestronk, Alan, Connolly, Anne M, Druley, Todd E, Weihl, Conrad C, Dobbs, Mathew B. \"Myosin binding protein C1: a novel gene for autosomal dominant distal arthrogryposis type 1.\" Human molecular genetics, 2010, pp. 1165-73, doi: https://doi.org/10.1093/hmg/ddp587', 'Chen, Zhe, Zhao, Tong-Jin, Li, Jie, Gao, Yan-Song, Meng, Fan-Guo, Yan, Yong-Bin, Zhou, Hai-Meng. \"Slow skeletal muscle myosin-binding protein-C (MyBPC1) mediates recruitment of muscle-type creatine kinase (CK) to myosin.\" The Biochemical journal, 2011, pp. 437-45, doi: https://doi.org/10.1042/BJ20102007', 'Gupta, Manish K, Robbins, Jeffrey. \"Post-translational control of cardiac hemodynamics through myosin binding protein C.\" Pflugers Archiv : European journal of physiology, 2014, pp. 231-6, doi: https://doi.org/10.1007/s00424-013-1377-y'], ['Yan, R, Shi, Q, Hu, X, Zhou, X. \"Reticulon proteins: emerging players in neurodegenerative diseases.\" Cellular and molecular life sciences : CMLS, 2006, pp. 877-89, doi: https://doi.org/10.1007/s00018-005-5338-2', 'Recabarren, Daniela, Alarcón, Marcelo. \"Gene networks in neurodegenerative disorders.\" Life sciences, 2017, pp. 83-97, doi: https://doi.org/10.1016/j.lfs.2017.06.009', 'Shi, Qi, Ge, Yingying, He, Wanxia, Hu, Xiangyou, Yan, Riqiang. \"RTN1 and RTN3 protein are differentially associated with senile plaques in Alzheimer\\'s brains.\" Scientific reports, 2017, pp. 6145, doi: https://doi.org/10.1038/s41598-017-05504-9'], ['Traini, Mathew, Quinn, Carmel M, Sandoval, Cecilia, Johansson, Erik, Schroder, Kate, Kockx, Maaike, Meikle, Peter J, Jessup, Wendy, Kritharides, Leonard. \"Sphingomyelin phosphodiesterase acid-like 3A (SMPDL3A) is a novel nucleotide phosphodiesterase regulated by cholesterol in human macrophages.\" The Journal of biological chemistry, 2014, pp. 32895-913, doi: https://doi.org/10.1074/jbc.M114.612341', 'Noto, Paul B, Bukhtiyarov, Yuri, Shi, Meng, McKeever, Brian M, McGeehan, Gerard M, Lala, Deepak S. \"Regulation of sphingomyelin phosphodiesterase acid-like 3A gene (SMPDL3A) by liver X receptors.\" Molecular pharmacology, 2012, pp. 719-27, doi: https://doi.org/10.1124/mol.112.078865', 'Traini, Mathew, Kumaran, Raani, Thaysen-Andersen, Morten, Kockx, Maaike, Jessup, Wendy, Kritharides, Leonard. \"N-glycosylation of human sphingomyelin phosphodiesterase acid-like 3A (SMPDL3A) is essential for stability, secretion and activity.\" The Biochemical journal, 2017, pp. 1071-1092, doi: https://doi.org/10.1042/BCJ20160735'], ['Eyvazi, Shirin, Kazemi, Bahram, Dastmalchi, Siavoush, Bandehpour, Mojgan. \"Involvement of CD24 in Multiple Cancer Related Pathways Makes It an Interesting New Target for Cancer Therapy.\" Current cancer drug targets, 2018, pp. 328-336, doi: https://doi.org/10.2174/1570163814666170818125036', 'Ni, Yang-Hong, Zhao, Xia, Wang, Wei. \"CD24, A Review of its Role in Tumor Diagnosis, Progression and Therapy.\" Current gene therapy, 2020, pp. 109-126, doi: https://doi.org/10.2174/1566523220666200623170738', 'Zhang, Yifan, Li, Bingjin, Zhang, Xingyi, Sonpavde, Guru P, Jiao, Kenneth, Zhang, Andrea, Zhang, Guangxin, Sun, Mei, Chu, Chengjing, Li, Feng, Wang, Lizhong, Cui, Ranji, Liu, Runhua. \"CD24 is a genetic modifier for risk and progression of prostate cancer.\" Molecular carcinogenesis, 2017, pp. 641-650, doi: https://doi.org/10.1002/mc.22522'], ['Gires, Olivier, Pan, Min, Schinke, Henrik, Canis, Martin, Baeuerle, Patrick A. \"Expression and function of epithelial cell adhesion molecule EpCAM: where are we after 40 years?\" Cancer metastasis reviews, 2020, pp. 969-987, doi: https://doi.org/10.1007/s10555-020-09898-3', 'Dollé, Laurent, Theise, Neil D, Schmelzer, Eva, Boulter, Luke, Gires, Olivier, van Grunsven, Leo A. \"EpCAM and the biology of hepatic stem/progenitor cells.\" American journal of physiology. Gastrointestinal and liver physiology, 2015, pp. G233-50, doi: https://doi.org/10.1152/ajpgi.00069.2014', 'Han, Jiyou, Won, Miae, Kim, Ji Hyeon, Jung, Eugeine, Min, Kyungim, Jangili, Paramesh, Kim, Jong Seung. \"Cancer stem cell-targeted bio-imaging and chemotherapeutic perspective.\" Chemical Society reviews, 2020, pp. 7856-7878, doi: https://doi.org/10.1039/d0cs00379d'], ['Chambers, Jennifer, Rabbitts, Terence H. \"LMO2 at 25 years: a paradigm of chromosomal translocation proteins.\" Open biology, 2015, pp. 150062, doi: https://doi.org/10.1098/rsob.150062', 'Kim, S-H, Kim, E-J, Hitomi, M, Oh, S-Y, Jin, X, Jeon, H-M, Beck, S, Jin, X, Kim, J-K, Park, C G, Chang, S-Y, Yin, J, Kim, T, Jeon, Y-J, Song, J, Lim, Y C, Lathia, J D, Nakano, I, Kim, H. \"The LIM-only transcription factor LMO2 determines tumorigenic and angiogenic traits in glioma stem cells.\" Cell death and differentiation, 2015, pp. 1517-25, doi: https://doi.org/10.1038/cdd.2015.7', 'Coma, Silvia, Allard-Ratick, Marc, Akino, Tomoshige, van Meeteren, Laurens A, Mammoto, Akiko, Klagsbrun, Michael. \"GATA2 and Lmo2 control angiogenesis and lymphangiogenesis via direct transcriptional regulation of neuropilin-2.\" Angiogenesis, 2013, pp. 939-52, doi: https://doi.org/10.1007/s10456-013-9370-9'], [], ['James, Nicole E, Gura, Megan, Woodman, Morgan, Freiman, Richard N, Ribeiro, Jennifer R. \"A bioinformatic analysis of WFDC2 (HE4) expression in high grade serous ovarian cancer reveals tumor-specific changes in metabolic and extracellular matrix gene expression.\" Medical oncology (Northwood, London, England), 2022, pp. 71, doi: https://doi.org/10.1007/s12032-022-01665-4', 'Lu, Jin, Yang, Yue, Yan, Kun, Deng, Zhuqin, Zhang, Haoxuan. \"[Expression and significance of WFDC2 in ovarian cancer based on multiple gene databases].\" Xi bao yu fen zi mian yi xue za zhi = Chinese journal of cellular and molecular immunology, 2018, pp. 528-534.'], []]\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "#####USE get_references_for_paragraph####\n",
    "saved_dict = {}\n",
    "i = 0\n",
    "for set_id, row in df.iterrows():\n",
    "#for i in range(startRow, df.shape[0]):\n",
    "#    row = df.iloc[i]\n",
    "    # if runOnlyExamples: # Only run examples\n",
    "    #    if df.iloc[i][nameCol] not in examplesTORun: \n",
    "    #        continue\n",
    "    if runVersion == \"initial\":\n",
    "        if df.loc[set_id, 'referenced_analysis'] is not None:\n",
    "            continue\n",
    "    if runVersion == \"additional\":\n",
    "        with open(toSaveFilePath + '.json') as f:\n",
    "            saved_dict = json.load(f)\n",
    "        if not (df.loc[set_id,'referenced_analysis'] == ''):\n",
    "            continue # skip this row because already done\n",
    "            \n",
    "    print('=========================================')\n",
    "    print('=========================================')\n",
    "    print('=========================================')\n",
    "\n",
    "    print(['dataframe row', set_id])\n",
    "    # check out the llm analysis \n",
    "    example_analysis = df.loc[set_id, LLM_analysisCol]\n",
    "    paragraphs = list(filter(lambda p: len(p.split()) > 5, example_analysis.split(\"\\n\")))\n",
    "    \n",
    "    try:\n",
    "        references, paragraph_dict = get_references_for_paragraphs(paragraphs, email = email, config =config, n=3, verbose=True, papers_query=20, return_paragraph_ref_data=True)\n",
    "        \n",
    "    except Exception as e:\n",
    "        print('Cannot get references for row', set_id, e)\n",
    "        references = ''\n",
    "        saved_dict[set_id] = None\n",
    "        \n",
    "    references.replace( '\\n', '')\n",
    "    \n",
    "    df.loc[set_id, 'referenced_analysis'] = references\n",
    "    saved_dict[set_id] = paragraph_dict\n",
    "    i += 1\n",
    "    if i%5==0:\n",
    "        df.to_csv(toSaveFilePath+ '.tsv', sep = '\\t')\n",
    "        with open(toSaveFilePath + '.json', 'w') as f:\n",
    "            json.dump(saved_dict, f)\n",
    "        \n",
    "# if not runOnlyExamples: \n",
    "df.to_csv(toSaveFilePath+'.tsv', sep = '\\t')\n",
    "with open(toSaveFilePath + '.json', 'w') as f:\n",
    "    json.dump(saved_dict, f)\n",
    "# check there is no None\n",
    "print(len(df[df['referenced_analysis'] == '']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(toSaveFilePath+'.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "est time per set:  70.13306418332186\n",
      "est. cost per set:  0.8977145454545454\n"
     ]
    }
   ],
   "source": [
    "with open('./logs/reference_checker_revision_test_11log.json', 'r') as log:\n",
    "    log_data = json.load(log)\n",
    "\n",
    "print('est time per set: ', log_data['time_taken_total']/11)\n",
    "print('est. cost per set: ', log_data['dollars_spent']/11)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "llm_eval",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
